// vue-speedometer/dist/vue-speedometer.es.js
function t(t18) {
  var e2 = typeof t18;
  return t18 != null && (e2 == "object" || e2 == "function");
}
var e = typeof global == "object" && global && global.Object === Object && global, n = typeof self == "object" && self && self.Object === Object && self, r = e || n || Function("return this")(), i = () => r.Date.now(), a = /\s/;
var o = /^\s+/;
function u(t18) {
  return t18 ? t18.slice(0, function(t19) {
    for (var e2 = t19.length; e2-- && a.test(t19.charAt(e2)); )
      ;
    return e2;
  }(t18) + 1).replace(o, "") : t18;
}
var s = r.Symbol, l = Object.prototype, c = l.hasOwnProperty, h = l.toString, f = s ? s.toStringTag : void 0;
var p = Object.prototype.toString;
var d = s ? s.toStringTag : void 0;
function v(t18) {
  return t18 == null ? t18 === void 0 ? "[object Undefined]" : "[object Null]" : d && d in Object(t18) ? function(t19) {
    var e2 = c.call(t19, f), n2 = t19[f];
    try {
      t19[f] = void 0;
      var r2 = true;
    } catch (t20) {
    }
    var i2 = h.call(t19);
    return r2 && (e2 ? t19[f] = n2 : delete t19[f]), i2;
  }(t18) : function(t19) {
    return p.call(t19);
  }(t18);
}
function g(t18) {
  return t18 != null && typeof t18 == "object";
}
function y(t18) {
  return typeof t18 == "symbol" || g(t18) && v(t18) == "[object Symbol]";
}
var m = /^[-+]0x[0-9a-f]+$/i, _ = /^0b[01]+$/i, b = /^0o[0-7]+$/i, w = parseInt;
function x(e2) {
  if (typeof e2 == "number")
    return e2;
  if (y(e2))
    return NaN;
  if (t(e2)) {
    var n2 = typeof e2.valueOf == "function" ? e2.valueOf() : e2;
    e2 = t(n2) ? n2 + "" : n2;
  }
  if (typeof e2 != "string")
    return e2 === 0 ? e2 : +e2;
  e2 = u(e2);
  var r2 = _.test(e2);
  return r2 || b.test(e2) ? w(e2.slice(2), r2 ? 2 : 8) : m.test(e2) ? NaN : +e2;
}
var M = Math.max, S = Math.min;
function A(e2, n2, r2) {
  var a2, o2, u2, s2, l2, c2, h2 = 0, f2 = false, p2 = false, d2 = true;
  if (typeof e2 != "function")
    throw new TypeError("Expected a function");
  function v2(t18) {
    var n3 = a2, r3 = o2;
    return a2 = o2 = void 0, h2 = t18, s2 = e2.apply(r3, n3);
  }
  function g2(t18) {
    return h2 = t18, l2 = setTimeout(m2, n2), f2 ? v2(t18) : s2;
  }
  function y2(t18) {
    var e3 = t18 - c2;
    return c2 === void 0 || e3 >= n2 || e3 < 0 || p2 && t18 - h2 >= u2;
  }
  function m2() {
    var t18 = i();
    if (y2(t18))
      return _2(t18);
    l2 = setTimeout(m2, function(t19) {
      var e3 = n2 - (t19 - c2);
      return p2 ? S(e3, u2 - (t19 - h2)) : e3;
    }(t18));
  }
  function _2(t18) {
    return l2 = void 0, d2 && a2 ? v2(t18) : (a2 = o2 = void 0, s2);
  }
  function b2() {
    var t18 = i(), e3 = y2(t18);
    if (a2 = arguments, o2 = this, c2 = t18, e3) {
      if (l2 === void 0)
        return g2(c2);
      if (p2)
        return clearTimeout(l2), l2 = setTimeout(m2, n2), v2(c2);
    }
    return l2 === void 0 && (l2 = setTimeout(m2, n2)), s2;
  }
  return n2 = x(n2) || 0, t(r2) && (f2 = !!r2.leading, u2 = (p2 = "maxWait" in r2) ? M(x(r2.maxWait) || 0, n2) : u2, d2 = "trailing" in r2 ? !!r2.trailing : d2), b2.cancel = function() {
    l2 !== void 0 && clearTimeout(l2), h2 = 0, a2 = c2 = o2 = l2 = void 0;
  }, b2.flush = function() {
    return l2 === void 0 ? s2 : _2(i());
  }, b2;
}
function N(t18, e2) {
  for (var n2 = -1, r2 = t18 == null ? 0 : t18.length; ++n2 < r2 && e2(t18[n2], n2, t18) !== false; )
    ;
  return t18;
}
var C, O = (t18, e2, n2) => {
  for (var r2 = -1, i2 = Object(t18), a2 = n2(t18), o2 = a2.length; o2--; ) {
    var u2 = a2[C ? o2 : ++r2];
    if (e2(i2[u2], u2, i2) === false)
      break;
  }
  return t18;
}, k = O;
function j(t18, e2) {
  for (var n2 = -1, r2 = Array(t18); ++n2 < t18; )
    r2[n2] = e2(n2);
  return r2;
}
function E(t18) {
  return g(t18) && v(t18) == "[object Arguments]";
}
var I = Object.prototype, T = I.hasOwnProperty, L = I.propertyIsEnumerable, P = E(function() {
  return arguments;
}()) ? E : (t18) => g(t18) && T.call(t18, "callee") && !L.call(t18, "callee"), R = Array.isArray;
var $ = typeof exports == "object" && exports && !exports.nodeType && exports, z = $ && typeof module == "object" && module && !module.nodeType && module, q = z && z.exports === $ ? r.Buffer : void 0, V = (q ? q.isBuffer : void 0) || function() {
  return false;
}, U = /^(?:0|[1-9]\d*)$/;
function F(t18, e2) {
  var n2 = typeof t18;
  return !!(e2 = e2 != null ? e2 : 9007199254740991) && (n2 == "number" || n2 != "symbol" && U.test(t18)) && t18 > -1 && t18 % 1 == 0 && t18 < e2;
}
function B(t18) {
  return typeof t18 == "number" && t18 > -1 && t18 % 1 == 0 && t18 <= 9007199254740991;
}
var D = {};
D["[object Float32Array]"] = D["[object Float64Array]"] = D["[object Int8Array]"] = D["[object Int16Array]"] = D["[object Int32Array]"] = D["[object Uint8Array]"] = D["[object Uint8ClampedArray]"] = D["[object Uint16Array]"] = D["[object Uint32Array]"] = true, D["[object Arguments]"] = D["[object Array]"] = D["[object ArrayBuffer]"] = D["[object Boolean]"] = D["[object DataView]"] = D["[object Date]"] = D["[object Error]"] = D["[object Function]"] = D["[object Map]"] = D["[object Number]"] = D["[object Object]"] = D["[object RegExp]"] = D["[object Set]"] = D["[object String]"] = D["[object WeakMap]"] = false;
var W, H = typeof exports == "object" && exports && !exports.nodeType && exports, X = H && typeof module == "object" && module && !module.nodeType && module, Y = X && X.exports === H && e.process, Q = (() => {
  try {
    var t18 = X && X.require && X.require("util").types;
    return t18 || Y && Y.binding && Y.binding("util");
  } catch (t19) {
  }
})(), G = Q && Q.isTypedArray, Z = G ? (W = G, (t18) => W(t18)) : function(t18) {
  return g(t18) && B(t18.length) && !!D[v(t18)];
}, J = Object.prototype.hasOwnProperty;
var K = Object.prototype;
function tt(t18) {
  var e2 = t18 && t18.constructor;
  return t18 === (typeof e2 == "function" && e2.prototype || K);
}
var et = function(t18, e2) {
  return (n2) => t18(e2(n2));
}(Object.keys, Object), nt = et, rt = Object.prototype.hasOwnProperty;
function it(t18) {
  if (!tt(t18))
    return nt(t18);
  var e2 = [];
  for (var n2 in Object(t18))
    rt.call(t18, n2) && n2 != "constructor" && e2.push(n2);
  return e2;
}
function at(e2) {
  if (!t(e2))
    return false;
  var n2 = v(e2);
  return n2 == "[object Function]" || n2 == "[object GeneratorFunction]" || n2 == "[object AsyncFunction]" || n2 == "[object Proxy]";
}
function ot(t18) {
  return t18 != null && B(t18.length) && !at(t18);
}
function ut(t18) {
  return ot(t18) ? function(t19, e2) {
    var n2 = R(t19), r2 = !n2 && P(t19), i2 = !n2 && !r2 && V(t19), a2 = !n2 && !r2 && !i2 && Z(t19), o2 = n2 || r2 || i2 || a2, u2 = o2 ? j(t19.length, String) : [], s2 = u2.length;
    for (var l2 in t19)
      !e2 && !J.call(t19, l2) || o2 && (l2 == "length" || i2 && (l2 == "offset" || l2 == "parent") || a2 && (l2 == "buffer" || l2 == "byteLength" || l2 == "byteOffset") || F(l2, s2)) || u2.push(l2);
    return u2;
  }(t18) : it(t18);
}
function st(t18, e2) {
  return t18 && k(t18, e2, ut);
}
var lt = function(t18, e2) {
  return (n2, r2) => {
    if (n2 == null)
      return n2;
    if (!ot(n2))
      return t18(n2, r2);
    for (var i2 = n2.length, a2 = e2 ? i2 : -1, o2 = Object(n2); (e2 ? a2-- : ++a2 < i2) && r2(o2[a2], a2, o2) !== false; )
      ;
    return n2;
  };
}(st), ct = lt;
function ht(t18) {
  return t18;
}
function ft(t18) {
  return typeof t18 == "function" ? t18 : ht;
}
function pt(t18, e2) {
  return (R(t18) ? N : ct)(t18, ft(e2));
}
function dt(t18, e2) {
  return t18 < e2 ? -1 : t18 > e2 ? 1 : t18 >= e2 ? 0 : NaN;
}
var vt, gt, yt = ((vt = dt).length === 1 && (gt = vt, vt = (t18, e2) => dt(gt(t18), e2)), { left: (t18, e2, n2, r2) => {
  for (n2 == null && (n2 = 0), r2 == null && (r2 = t18.length); n2 < r2; ) {
    var i2 = n2 + r2 >>> 1;
    vt(t18[i2], e2) < 0 ? n2 = i2 + 1 : r2 = i2;
  }
  return n2;
}, right: (t18, e2, n2, r2) => {
  for (n2 == null && (n2 = 0), r2 == null && (r2 = t18.length); n2 < r2; ) {
    var i2 = n2 + r2 >>> 1;
    vt(t18[i2], e2) > 0 ? r2 = i2 : n2 = i2 + 1;
  }
  return n2;
} }), mt = yt.right;
var _t = Math.sqrt(50), bt = Math.sqrt(10), wt = Math.sqrt(2);
function xt(t18, e2, n2) {
  var r2 = (e2 - t18) / Math.max(0, n2), i2 = Math.floor(Math.log(r2) / Math.LN10), a2 = r2 / Math.pow(10, i2);
  return i2 >= 0 ? (a2 >= _t ? 10 : a2 >= bt ? 5 : a2 >= wt ? 2 : 1) * Math.pow(10, i2) : -Math.pow(10, -i2) / (a2 >= _t ? 10 : a2 >= bt ? 5 : a2 >= wt ? 2 : 1);
}
var Mt = { value: () => {
} };
function St() {
  for (var t18, e2 = 0, n2 = arguments.length, r2 = {}; e2 < n2; ++e2) {
    if (!(t18 = arguments[e2] + "") || t18 in r2 || /[\s.]/.test(t18))
      throw new Error("illegal type: " + t18);
    r2[t18] = [];
  }
  return new At(r2);
}
function At(t18) {
  this._ = t18;
}
function Nt(t18, e2) {
  return t18.trim().split(/^|\s+/).map((t19) => {
    var n2 = "", r2 = t19.indexOf(".");
    if (r2 >= 0 && (n2 = t19.slice(r2 + 1), t19 = t19.slice(0, r2)), t19 && !e2.hasOwnProperty(t19))
      throw new Error("unknown type: " + t19);
    return { type: t19, name: n2 };
  });
}
function Ct(t18, e2) {
  for (var n2, r2 = 0, i2 = t18.length; r2 < i2; ++r2)
    if ((n2 = t18[r2]).name === e2)
      return n2.value;
}
function Ot(t18, e2, n2) {
  for (var r2 = 0, i2 = t18.length; r2 < i2; ++r2)
    if (t18[r2].name === e2) {
      t18[r2] = Mt, t18 = t18.slice(0, r2).concat(t18.slice(r2 + 1));
      break;
    }
  return n2 != null && t18.push({ name: e2, value: n2 }), t18;
}
At.prototype = St.prototype = { constructor: At, on: function(t18, e2) {
  var n2, r2 = this._, i2 = Nt(t18 + "", r2), a2 = -1, o2 = i2.length;
  if (!(arguments.length < 2)) {
    if (e2 != null && typeof e2 != "function")
      throw new Error("invalid callback: " + e2);
    for (; ++a2 < o2; )
      if (n2 = (t18 = i2[a2]).type)
        r2[n2] = Ot(r2[n2], t18.name, e2);
      else if (e2 == null)
        for (n2 in r2)
          r2[n2] = Ot(r2[n2], t18.name, null);
    return this;
  }
  for (; ++a2 < o2; )
    if ((n2 = (t18 = i2[a2]).type) && (n2 = Ct(r2[n2], t18.name)))
      return n2;
}, copy: function() {
  var t18 = {}, e2 = this._;
  for (var n2 in e2)
    t18[n2] = e2[n2].slice();
  return new At(t18);
}, call: function(t18, e2) {
  if ((n2 = arguments.length - 2) > 0)
    for (var n2, r2, i2 = new Array(n2), a2 = 0; a2 < n2; ++a2)
      i2[a2] = arguments[a2 + 2];
  if (!this._.hasOwnProperty(t18))
    throw new Error("unknown type: " + t18);
  for (a2 = 0, n2 = (r2 = this._[t18]).length; a2 < n2; ++a2)
    r2[a2].value.apply(e2, i2);
}, apply: function(t18, e2, n2) {
  if (!this._.hasOwnProperty(t18))
    throw new Error("unknown type: " + t18);
  for (var r2 = this._[t18], i2 = 0, a2 = r2.length; i2 < a2; ++i2)
    r2[i2].value.apply(e2, n2);
} };
var kt = "http://www.w3.org/1999/xhtml", jt = { svg: "http://www.w3.org/2000/svg", xhtml: kt, xlink: "http://www.w3.org/1999/xlink", xml: "http://www.w3.org/XML/1998/namespace", xmlns: "http://www.w3.org/2000/xmlns/" };
function Et(t18) {
  var e2 = t18 += "", n2 = e2.indexOf(":");
  return n2 >= 0 && (e2 = t18.slice(0, n2)) !== "xmlns" && (t18 = t18.slice(n2 + 1)), jt.hasOwnProperty(e2) ? { space: jt[e2], local: t18 } : t18;
}
function It(t18) {
  return function() {
    var e2 = this.ownerDocument, n2 = this.namespaceURI;
    return n2 === kt && e2.documentElement.namespaceURI === kt ? e2.createElement(t18) : e2.createElementNS(n2, t18);
  };
}
function Tt(t18) {
  return function() {
    return this.ownerDocument.createElementNS(t18.space, t18.local);
  };
}
function Lt(t18) {
  var e2 = Et(t18);
  return (e2.local ? Tt : It)(e2);
}
function Pt() {
}
function Rt(t18) {
  return t18 == null ? Pt : function() {
    return this.querySelector(t18);
  };
}
function $t() {
  return [];
}
function zt(t18) {
  return t18 == null ? $t : function() {
    return this.querySelectorAll(t18);
  };
}
function qt(t18) {
  return function() {
    return this.matches(t18);
  };
}
function Vt(t18) {
  return new Array(t18.length);
}
function Ut(t18, e2) {
  this.ownerDocument = t18.ownerDocument, this.namespaceURI = t18.namespaceURI, this._next = null, this._parent = t18, this.__data__ = e2;
}
Ut.prototype = { constructor: Ut, appendChild: function(t18) {
  return this._parent.insertBefore(t18, this._next);
}, insertBefore: function(t18, e2) {
  return this._parent.insertBefore(t18, e2);
}, querySelector: function(t18) {
  return this._parent.querySelector(t18);
}, querySelectorAll: function(t18) {
  return this._parent.querySelectorAll(t18);
} };
function Ft(t18, e2, n2, r2, i2, a2) {
  for (var o2, u2 = 0, s2 = e2.length, l2 = a2.length; u2 < l2; ++u2)
    (o2 = e2[u2]) ? (o2.__data__ = a2[u2], r2[u2] = o2) : n2[u2] = new Ut(t18, a2[u2]);
  for (; u2 < s2; ++u2)
    (o2 = e2[u2]) && (i2[u2] = o2);
}
function Bt(t18, e2, n2, r2, i2, a2, o2) {
  var u2, s2, l2, c2 = {}, h2 = e2.length, f2 = a2.length, p2 = new Array(h2);
  for (u2 = 0; u2 < h2; ++u2)
    (s2 = e2[u2]) && (p2[u2] = l2 = "$" + o2.call(s2, s2.__data__, u2, e2), l2 in c2 ? i2[u2] = s2 : c2[l2] = s2);
  for (u2 = 0; u2 < f2; ++u2)
    (s2 = c2[l2 = "$" + o2.call(t18, a2[u2], u2, a2)]) ? (r2[u2] = s2, s2.__data__ = a2[u2], c2[l2] = null) : n2[u2] = new Ut(t18, a2[u2]);
  for (u2 = 0; u2 < h2; ++u2)
    (s2 = e2[u2]) && c2[p2[u2]] === s2 && (i2[u2] = s2);
}
function Dt(t18, e2) {
  return t18 < e2 ? -1 : t18 > e2 ? 1 : t18 >= e2 ? 0 : NaN;
}
function Wt(t18) {
  return function() {
    this.removeAttribute(t18);
  };
}
function Ht(t18) {
  return function() {
    this.removeAttributeNS(t18.space, t18.local);
  };
}
function Xt(t18, e2) {
  return function() {
    this.setAttribute(t18, e2);
  };
}
function Yt(t18, e2) {
  return function() {
    this.setAttributeNS(t18.space, t18.local, e2);
  };
}
function Qt(t18, e2) {
  return function() {
    var n2 = e2.apply(this, arguments);
    n2 == null ? this.removeAttribute(t18) : this.setAttribute(t18, n2);
  };
}
function Gt(t18, e2) {
  return function() {
    var n2 = e2.apply(this, arguments);
    n2 == null ? this.removeAttributeNS(t18.space, t18.local) : this.setAttributeNS(t18.space, t18.local, n2);
  };
}
function Zt(t18) {
  return t18.ownerDocument && t18.ownerDocument.defaultView || t18.document && t18 || t18.defaultView;
}
function Jt(t18) {
  return function() {
    this.style.removeProperty(t18);
  };
}
function Kt(t18, e2, n2) {
  return function() {
    this.style.setProperty(t18, e2, n2);
  };
}
function te(t18, e2, n2) {
  return function() {
    var r2 = e2.apply(this, arguments);
    r2 == null ? this.style.removeProperty(t18) : this.style.setProperty(t18, r2, n2);
  };
}
function ee(t18, e2) {
  return t18.style.getPropertyValue(e2) || Zt(t18).getComputedStyle(t18, null).getPropertyValue(e2);
}
function ne(t18) {
  return function() {
    delete this[t18];
  };
}
function re(t18, e2) {
  return function() {
    this[t18] = e2;
  };
}
function ie(t18, e2) {
  return function() {
    var n2 = e2.apply(this, arguments);
    n2 == null ? delete this[t18] : this[t18] = n2;
  };
}
function ae(t18) {
  return t18.trim().split(/^|\s+/);
}
function oe(t18) {
  return t18.classList || new ue(t18);
}
function ue(t18) {
  this._node = t18, this._names = ae(t18.getAttribute("class") || "");
}
function se(t18, e2) {
  for (var n2 = oe(t18), r2 = -1, i2 = e2.length; ++r2 < i2; )
    n2.add(e2[r2]);
}
function le(t18, e2) {
  for (var n2 = oe(t18), r2 = -1, i2 = e2.length; ++r2 < i2; )
    n2.remove(e2[r2]);
}
function ce(t18) {
  return function() {
    se(this, t18);
  };
}
function he(t18) {
  return function() {
    le(this, t18);
  };
}
function fe(t18, e2) {
  return function() {
    (e2.apply(this, arguments) ? se : le)(this, t18);
  };
}
function pe() {
  this.textContent = "";
}
function de(t18) {
  return function() {
    this.textContent = t18;
  };
}
function ve(t18) {
  return function() {
    var e2 = t18.apply(this, arguments);
    this.textContent = e2 != null ? e2 : "";
  };
}
function ge() {
  this.innerHTML = "";
}
function ye(t18) {
  return function() {
    this.innerHTML = t18;
  };
}
function me(t18) {
  return function() {
    var e2 = t18.apply(this, arguments);
    this.innerHTML = e2 != null ? e2 : "";
  };
}
function _e() {
  this.nextSibling && this.parentNode.appendChild(this);
}
function be() {
  this.previousSibling && this.parentNode.insertBefore(this, this.parentNode.firstChild);
}
function we() {
  return null;
}
function xe() {
  var t18 = this.parentNode;
  t18 && t18.removeChild(this);
}
function Me() {
  var t18 = this.cloneNode(false), e2 = this.parentNode;
  return e2 ? e2.insertBefore(t18, this.nextSibling) : t18;
}
function Se() {
  var t18 = this.cloneNode(true), e2 = this.parentNode;
  return e2 ? e2.insertBefore(t18, this.nextSibling) : t18;
}
ue.prototype = { add: function(t18) {
  this._names.indexOf(t18) < 0 && (this._names.push(t18), this._node.setAttribute("class", this._names.join(" ")));
}, remove: function(t18) {
  var e2 = this._names.indexOf(t18);
  e2 >= 0 && (this._names.splice(e2, 1), this._node.setAttribute("class", this._names.join(" ")));
}, contains: function(t18) {
  return this._names.indexOf(t18) >= 0;
} };
var Ae = {};
typeof document != "undefined" && ("onmouseenter" in document.documentElement || (Ae = { mouseenter: "mouseover", mouseleave: "mouseout" }));
function Ne(t18, e2, n2) {
  return t18 = Ce(t18, e2, n2), function(e3) {
    var n3 = e3.relatedTarget;
    n3 && (n3 === this || 8 & n3.compareDocumentPosition(this)) || t18.call(this, e3);
  };
}
function Ce(t18, e2, n2) {
  return function(r2) {
    try {
      t18.call(this, this.__data__, e2, n2);
    } finally {
    }
  };
}
function Oe(t18) {
  return t18.trim().split(/^|\s+/).map((t19) => {
    var e2 = "", n2 = t19.indexOf(".");
    return n2 >= 0 && (e2 = t19.slice(n2 + 1), t19 = t19.slice(0, n2)), { type: t19, name: e2 };
  });
}
function ke(t18) {
  return function() {
    var e2 = this.__on;
    if (e2) {
      for (var n2, r2 = 0, i2 = -1, a2 = e2.length; r2 < a2; ++r2)
        n2 = e2[r2], t18.type && n2.type !== t18.type || n2.name !== t18.name ? e2[++i2] = n2 : this.removeEventListener(n2.type, n2.listener, n2.capture);
      ++i2 ? e2.length = i2 : delete this.__on;
    }
  };
}
function je(t18, e2, n2) {
  var r2 = Ae.hasOwnProperty(t18.type) ? Ne : Ce;
  return function(i2, a2, o2) {
    var u2, s2 = this.__on, l2 = r2(e2, a2, o2);
    if (s2) {
      for (var c2 = 0, h2 = s2.length; c2 < h2; ++c2)
        if ((u2 = s2[c2]).type === t18.type && u2.name === t18.name)
          return this.removeEventListener(u2.type, u2.listener, u2.capture), this.addEventListener(u2.type, u2.listener = l2, u2.capture = n2), void (u2.value = e2);
    }
    this.addEventListener(t18.type, l2, n2), u2 = { type: t18.type, name: t18.name, value: e2, listener: l2, capture: n2 }, s2 ? s2.push(u2) : this.__on = [u2];
  };
}
function Ee(t18, e2, n2) {
  var r2 = Zt(t18), i2 = r2.CustomEvent;
  typeof i2 == "function" ? i2 = new i2(e2, n2) : (i2 = r2.document.createEvent("Event"), n2 ? (i2.initEvent(e2, n2.bubbles, n2.cancelable), i2.detail = n2.detail) : i2.initEvent(e2, false, false)), t18.dispatchEvent(i2);
}
function Ie(t18, e2) {
  return function() {
    return Ee(this, t18, e2);
  };
}
function Te(t18, e2) {
  return function() {
    return Ee(this, t18, e2.apply(this, arguments));
  };
}
var Le = [null];
function Pe(t18, e2) {
  this._groups = t18, this._parents = e2;
}
function Re() {
  return new Pe([[document.documentElement]], Le);
}
function $e(t18) {
  return typeof t18 == "string" ? new Pe([[document.querySelector(t18)]], [document.documentElement]) : new Pe([[t18]], Le);
}
function ze(t18, e2, n2) {
  t18.prototype = e2.prototype = n2, n2.constructor = t18;
}
function qe(t18, e2) {
  var n2 = Object.create(t18.prototype);
  for (var r2 in e2)
    n2[r2] = e2[r2];
  return n2;
}
function Ve() {
}
Pe.prototype = Re.prototype = { constructor: Pe, select: function(t18) {
  typeof t18 != "function" && (t18 = Rt(t18));
  for (var e2 = this._groups, n2 = e2.length, r2 = new Array(n2), i2 = 0; i2 < n2; ++i2)
    for (var a2, o2, u2 = e2[i2], s2 = u2.length, l2 = r2[i2] = new Array(s2), c2 = 0; c2 < s2; ++c2)
      (a2 = u2[c2]) && (o2 = t18.call(a2, a2.__data__, c2, u2)) && ("__data__" in a2 && (o2.__data__ = a2.__data__), l2[c2] = o2);
  return new Pe(r2, this._parents);
}, selectAll: function(t18) {
  typeof t18 != "function" && (t18 = zt(t18));
  for (var e2 = this._groups, n2 = e2.length, r2 = [], i2 = [], a2 = 0; a2 < n2; ++a2)
    for (var o2, u2 = e2[a2], s2 = u2.length, l2 = 0; l2 < s2; ++l2)
      (o2 = u2[l2]) && (r2.push(t18.call(o2, o2.__data__, l2, u2)), i2.push(o2));
  return new Pe(r2, i2);
}, filter: function(t18) {
  typeof t18 != "function" && (t18 = qt(t18));
  for (var e2 = this._groups, n2 = e2.length, r2 = new Array(n2), i2 = 0; i2 < n2; ++i2)
    for (var a2, o2 = e2[i2], u2 = o2.length, s2 = r2[i2] = [], l2 = 0; l2 < u2; ++l2)
      (a2 = o2[l2]) && t18.call(a2, a2.__data__, l2, o2) && s2.push(a2);
  return new Pe(r2, this._parents);
}, data: function(t18, e2) {
  if (!t18)
    return p2 = new Array(this.size()), l2 = -1, this.each((t19) => {
      p2[++l2] = t19;
    }), p2;
  var n2 = e2 ? Bt : Ft, r2 = this._parents, i2 = this._groups;
  typeof t18 != "function" && (t18 = function(t19) {
    return () => t19;
  }(t18));
  for (var a2 = i2.length, o2 = new Array(a2), u2 = new Array(a2), s2 = new Array(a2), l2 = 0; l2 < a2; ++l2) {
    var c2 = r2[l2], h2 = i2[l2], f2 = h2.length, p2 = t18.call(c2, c2 && c2.__data__, l2, r2), d2 = p2.length, v2 = u2[l2] = new Array(d2), g2 = o2[l2] = new Array(d2);
    n2(c2, h2, v2, g2, s2[l2] = new Array(f2), p2, e2);
    for (var y2, m2, _2 = 0, b2 = 0; _2 < d2; ++_2)
      if (y2 = v2[_2]) {
        for (_2 >= b2 && (b2 = _2 + 1); !(m2 = g2[b2]) && ++b2 < d2; )
          ;
        y2._next = m2 || null;
      }
  }
  return (o2 = new Pe(o2, r2))._enter = u2, o2._exit = s2, o2;
}, enter: function() {
  return new Pe(this._enter || this._groups.map(Vt), this._parents);
}, exit: function() {
  return new Pe(this._exit || this._groups.map(Vt), this._parents);
}, join: function(t18, e2, n2) {
  var r2 = this.enter(), i2 = this, a2 = this.exit();
  return r2 = typeof t18 == "function" ? t18(r2) : r2.append(t18 + ""), e2 != null && (i2 = e2(i2)), n2 == null ? a2.remove() : n2(a2), r2 && i2 ? r2.merge(i2).order() : i2;
}, merge: function(t18) {
  for (var e2 = this._groups, n2 = t18._groups, r2 = e2.length, i2 = n2.length, a2 = Math.min(r2, i2), o2 = new Array(r2), u2 = 0; u2 < a2; ++u2)
    for (var s2, l2 = e2[u2], c2 = n2[u2], h2 = l2.length, f2 = o2[u2] = new Array(h2), p2 = 0; p2 < h2; ++p2)
      (s2 = l2[p2] || c2[p2]) && (f2[p2] = s2);
  for (; u2 < r2; ++u2)
    o2[u2] = e2[u2];
  return new Pe(o2, this._parents);
}, order: function() {
  for (var t18 = this._groups, e2 = -1, n2 = t18.length; ++e2 < n2; )
    for (var r2, i2 = t18[e2], a2 = i2.length - 1, o2 = i2[a2]; --a2 >= 0; )
      (r2 = i2[a2]) && (o2 && 4 ^ r2.compareDocumentPosition(o2) && o2.parentNode.insertBefore(r2, o2), o2 = r2);
  return this;
}, sort: function(t18) {
  function e2(e3, n3) {
    return e3 && n3 ? t18(e3.__data__, n3.__data__) : !e3 - !n3;
  }
  t18 || (t18 = Dt);
  for (var n2 = this._groups, r2 = n2.length, i2 = new Array(r2), a2 = 0; a2 < r2; ++a2) {
    for (var o2, u2 = n2[a2], s2 = u2.length, l2 = i2[a2] = new Array(s2), c2 = 0; c2 < s2; ++c2)
      (o2 = u2[c2]) && (l2[c2] = o2);
    l2.sort(e2);
  }
  return new Pe(i2, this._parents).order();
}, call: function() {
  var t18 = arguments[0];
  return arguments[0] = this, t18.apply(null, arguments), this;
}, nodes: function() {
  var t18 = new Array(this.size()), e2 = -1;
  return this.each(function() {
    t18[++e2] = this;
  }), t18;
}, node: function() {
  for (var t18 = this._groups, e2 = 0, n2 = t18.length; e2 < n2; ++e2)
    for (var r2 = t18[e2], i2 = 0, a2 = r2.length; i2 < a2; ++i2) {
      var o2 = r2[i2];
      if (o2)
        return o2;
    }
  return null;
}, size: function() {
  var t18 = 0;
  return this.each(() => {
    ++t18;
  }), t18;
}, empty: function() {
  return !this.node();
}, each: function(t18) {
  for (var e2 = this._groups, n2 = 0, r2 = e2.length; n2 < r2; ++n2)
    for (var i2, a2 = e2[n2], o2 = 0, u2 = a2.length; o2 < u2; ++o2)
      (i2 = a2[o2]) && t18.call(i2, i2.__data__, o2, a2);
  return this;
}, attr: function(t18, e2) {
  var n2 = Et(t18);
  if (arguments.length < 2) {
    var r2 = this.node();
    return n2.local ? r2.getAttributeNS(n2.space, n2.local) : r2.getAttribute(n2);
  }
  return this.each((e2 == null ? n2.local ? Ht : Wt : typeof e2 == "function" ? n2.local ? Gt : Qt : n2.local ? Yt : Xt)(n2, e2));
}, style: function(t18, e2, n2) {
  return arguments.length > 1 ? this.each((e2 == null ? Jt : typeof e2 == "function" ? te : Kt)(t18, e2, n2 != null ? n2 : "")) : ee(this.node(), t18);
}, property: function(t18, e2) {
  return arguments.length > 1 ? this.each((e2 == null ? ne : typeof e2 == "function" ? ie : re)(t18, e2)) : this.node()[t18];
}, classed: function(t18, e2) {
  var n2 = ae(t18 + "");
  if (arguments.length < 2) {
    for (var r2 = oe(this.node()), i2 = -1, a2 = n2.length; ++i2 < a2; )
      if (!r2.contains(n2[i2]))
        return false;
    return true;
  }
  return this.each((typeof e2 == "function" ? fe : e2 ? ce : he)(n2, e2));
}, text: function(t18) {
  return arguments.length ? this.each(t18 == null ? pe : (typeof t18 == "function" ? ve : de)(t18)) : this.node().textContent;
}, html: function(t18) {
  return arguments.length ? this.each(t18 == null ? ge : (typeof t18 == "function" ? me : ye)(t18)) : this.node().innerHTML;
}, raise: function() {
  return this.each(_e);
}, lower: function() {
  return this.each(be);
}, append: function(t18) {
  var e2 = typeof t18 == "function" ? t18 : Lt(t18);
  return this.select(function() {
    return this.appendChild(e2.apply(this, arguments));
  });
}, insert: function(t18, e2) {
  var n2 = typeof t18 == "function" ? t18 : Lt(t18), r2 = e2 == null ? we : typeof e2 == "function" ? e2 : Rt(e2);
  return this.select(function() {
    return this.insertBefore(n2.apply(this, arguments), r2.apply(this, arguments) || null);
  });
}, remove: function() {
  return this.each(xe);
}, clone: function(t18) {
  return this.select(t18 ? Se : Me);
}, datum: function(t18) {
  return arguments.length ? this.property("__data__", t18) : this.node().__data__;
}, on: function(t18, e2, n2) {
  var r2, i2, a2 = Oe(t18 + ""), o2 = a2.length;
  if (!(arguments.length < 2)) {
    for (u2 = e2 ? je : ke, n2 == null && (n2 = false), r2 = 0; r2 < o2; ++r2)
      this.each(u2(a2[r2], e2, n2));
    return this;
  }
  var u2 = this.node().__on;
  if (u2) {
    for (var s2, l2 = 0, c2 = u2.length; l2 < c2; ++l2)
      for (r2 = 0, s2 = u2[l2]; r2 < o2; ++r2)
        if ((i2 = a2[r2]).type === s2.type && i2.name === s2.name)
          return s2.value;
  }
}, dispatch: function(t18, e2) {
  return this.each((typeof e2 == "function" ? Te : Ie)(t18, e2));
} };
var Ue = 1 / 0.7, Fe = "\\s*([+-]?\\d+)\\s*", Be = "\\s*([+-]?\\d*\\.?\\d+(?:[eE][+-]?\\d+)?)\\s*", De = "\\s*([+-]?\\d*\\.?\\d+(?:[eE][+-]?\\d+)?)%\\s*", We = /^#([0-9a-f]{3,8})$/, He = new RegExp("^rgb\\(" + [Fe, Fe, Fe] + "\\)$"), Xe = new RegExp("^rgb\\(" + [De, De, De] + "\\)$"), Ye = new RegExp("^rgba\\(" + [Fe, Fe, Fe, Be] + "\\)$"), Qe = new RegExp("^rgba\\(" + [De, De, De, Be] + "\\)$"), Ge = new RegExp("^hsl\\(" + [Be, De, De] + "\\)$"), Ze = new RegExp("^hsla\\(" + [Be, De, De, Be] + "\\)$"), Je = { aliceblue: 15792383, antiquewhite: 16444375, aqua: 65535, aquamarine: 8388564, azure: 15794175, beige: 16119260, bisque: 16770244, black: 0, blanchedalmond: 16772045, blue: 255, blueviolet: 9055202, brown: 10824234, burlywood: 14596231, cadetblue: 6266528, chartreuse: 8388352, chocolate: 13789470, coral: 16744272, cornflowerblue: 6591981, cornsilk: 16775388, crimson: 14423100, cyan: 65535, darkblue: 139, darkcyan: 35723, darkgoldenrod: 12092939, darkgray: 11119017, darkgreen: 25600, darkgrey: 11119017, darkkhaki: 12433259, darkmagenta: 9109643, darkolivegreen: 5597999, darkorange: 16747520, darkorchid: 10040012, darkred: 9109504, darksalmon: 15308410, darkseagreen: 9419919, darkslateblue: 4734347, darkslategray: 3100495, darkslategrey: 3100495, darkturquoise: 52945, darkviolet: 9699539, deeppink: 16716947, deepskyblue: 49151, dimgray: 6908265, dimgrey: 6908265, dodgerblue: 2003199, firebrick: 11674146, floralwhite: 16775920, forestgreen: 2263842, fuchsia: 16711935, gainsboro: 14474460, ghostwhite: 16316671, gold: 16766720, goldenrod: 14329120, gray: 8421504, green: 32768, greenyellow: 11403055, grey: 8421504, honeydew: 15794160, hotpink: 16738740, indianred: 13458524, indigo: 4915330, ivory: 16777200, khaki: 15787660, lavender: 15132410, lavenderblush: 16773365, lawngreen: 8190976, lemonchiffon: 16775885, lightblue: 11393254, lightcoral: 15761536, lightcyan: 14745599, lightgoldenrodyellow: 16448210, lightgray: 13882323, lightgreen: 9498256, lightgrey: 13882323, lightpink: 16758465, lightsalmon: 16752762, lightseagreen: 2142890, lightskyblue: 8900346, lightslategray: 7833753, lightslategrey: 7833753, lightsteelblue: 11584734, lightyellow: 16777184, lime: 65280, limegreen: 3329330, linen: 16445670, magenta: 16711935, maroon: 8388608, mediumaquamarine: 6737322, mediumblue: 205, mediumorchid: 12211667, mediumpurple: 9662683, mediumseagreen: 3978097, mediumslateblue: 8087790, mediumspringgreen: 64154, mediumturquoise: 4772300, mediumvioletred: 13047173, midnightblue: 1644912, mintcream: 16121850, mistyrose: 16770273, moccasin: 16770229, navajowhite: 16768685, navy: 128, oldlace: 16643558, olive: 8421376, olivedrab: 7048739, orange: 16753920, orangered: 16729344, orchid: 14315734, palegoldenrod: 15657130, palegreen: 10025880, paleturquoise: 11529966, palevioletred: 14381203, papayawhip: 16773077, peachpuff: 16767673, peru: 13468991, pink: 16761035, plum: 14524637, powderblue: 11591910, purple: 8388736, rebeccapurple: 6697881, red: 16711680, rosybrown: 12357519, royalblue: 4286945, saddlebrown: 9127187, salmon: 16416882, sandybrown: 16032864, seagreen: 3050327, seashell: 16774638, sienna: 10506797, silver: 12632256, skyblue: 8900331, slateblue: 6970061, slategray: 7372944, slategrey: 7372944, snow: 16775930, springgreen: 65407, steelblue: 4620980, tan: 13808780, teal: 32896, thistle: 14204888, tomato: 16737095, turquoise: 4251856, violet: 15631086, wheat: 16113331, white: 16777215, whitesmoke: 16119285, yellow: 16776960, yellowgreen: 10145074 };
function Ke() {
  return this.rgb().formatHex();
}
function tn() {
  return this.rgb().formatRgb();
}
function en(t18) {
  var e2, n2;
  return t18 = (t18 + "").trim().toLowerCase(), (e2 = We.exec(t18)) ? (n2 = e2[1].length, e2 = parseInt(e2[1], 16), n2 === 6 ? nn(e2) : n2 === 3 ? new un(e2 >> 8 & 15 | e2 >> 4 & 240, e2 >> 4 & 15 | 240 & e2, (15 & e2) << 4 | 15 & e2, 1) : n2 === 8 ? rn(e2 >> 24 & 255, e2 >> 16 & 255, e2 >> 8 & 255, (255 & e2) / 255) : n2 === 4 ? rn(e2 >> 12 & 15 | e2 >> 8 & 240, e2 >> 8 & 15 | e2 >> 4 & 240, e2 >> 4 & 15 | 240 & e2, ((15 & e2) << 4 | 15 & e2) / 255) : null) : (e2 = He.exec(t18)) ? new un(e2[1], e2[2], e2[3], 1) : (e2 = Xe.exec(t18)) ? new un(255 * e2[1] / 100, 255 * e2[2] / 100, 255 * e2[3] / 100, 1) : (e2 = Ye.exec(t18)) ? rn(e2[1], e2[2], e2[3], e2[4]) : (e2 = Qe.exec(t18)) ? rn(255 * e2[1] / 100, 255 * e2[2] / 100, 255 * e2[3] / 100, e2[4]) : (e2 = Ge.exec(t18)) ? hn(e2[1], e2[2] / 100, e2[3] / 100, 1) : (e2 = Ze.exec(t18)) ? hn(e2[1], e2[2] / 100, e2[3] / 100, e2[4]) : Je.hasOwnProperty(t18) ? nn(Je[t18]) : t18 === "transparent" ? new un(NaN, NaN, NaN, 0) : null;
}
function nn(t18) {
  return new un(t18 >> 16 & 255, t18 >> 8 & 255, 255 & t18, 1);
}
function rn(t18, e2, n2, r2) {
  return r2 <= 0 && (t18 = e2 = n2 = NaN), new un(t18, e2, n2, r2);
}
function an(t18) {
  return t18 instanceof Ve || (t18 = en(t18)), t18 ? new un((t18 = t18.rgb()).r, t18.g, t18.b, t18.opacity) : new un();
}
function on(t18, e2, n2, r2) {
  return arguments.length === 1 ? an(t18) : new un(t18, e2, n2, r2 != null ? r2 : 1);
}
function un(t18, e2, n2, r2) {
  this.r = +t18, this.g = +e2, this.b = +n2, this.opacity = +r2;
}
function sn() {
  return "#" + cn(this.r) + cn(this.g) + cn(this.b);
}
function ln() {
  var t18 = this.opacity;
  return ((t18 = isNaN(t18) ? 1 : Math.max(0, Math.min(1, t18))) === 1 ? "rgb(" : "rgba(") + Math.max(0, Math.min(255, Math.round(this.r) || 0)) + ", " + Math.max(0, Math.min(255, Math.round(this.g) || 0)) + ", " + Math.max(0, Math.min(255, Math.round(this.b) || 0)) + (t18 === 1 ? ")" : ", " + t18 + ")");
}
function cn(t18) {
  return ((t18 = Math.max(0, Math.min(255, Math.round(t18) || 0))) < 16 ? "0" : "") + t18.toString(16);
}
function hn(t18, e2, n2, r2) {
  return r2 <= 0 ? t18 = e2 = n2 = NaN : n2 <= 0 || n2 >= 1 ? t18 = e2 = NaN : e2 <= 0 && (t18 = NaN), new dn(t18, e2, n2, r2);
}
function fn(t18) {
  if (t18 instanceof dn)
    return new dn(t18.h, t18.s, t18.l, t18.opacity);
  if (t18 instanceof Ve || (t18 = en(t18)), !t18)
    return new dn();
  if (t18 instanceof dn)
    return t18;
  var e2 = (t18 = t18.rgb()).r / 255, n2 = t18.g / 255, r2 = t18.b / 255, i2 = Math.min(e2, n2, r2), a2 = Math.max(e2, n2, r2), o2 = NaN, u2 = a2 - i2, s2 = (a2 + i2) / 2;
  return u2 ? (o2 = e2 === a2 ? (n2 - r2) / u2 + 6 * (n2 < r2) : n2 === a2 ? (r2 - e2) / u2 + 2 : (e2 - n2) / u2 + 4, u2 /= s2 < 0.5 ? a2 + i2 : 2 - a2 - i2, o2 *= 60) : u2 = s2 > 0 && s2 < 1 ? 0 : o2, new dn(o2, u2, s2, t18.opacity);
}
function pn(t18, e2, n2, r2) {
  return arguments.length === 1 ? fn(t18) : new dn(t18, e2, n2, r2 != null ? r2 : 1);
}
function dn(t18, e2, n2, r2) {
  this.h = +t18, this.s = +e2, this.l = +n2, this.opacity = +r2;
}
function vn(t18, e2, n2) {
  return 255 * (t18 < 60 ? e2 + (n2 - e2) * t18 / 60 : t18 < 180 ? n2 : t18 < 240 ? e2 + (n2 - e2) * (240 - t18) / 60 : e2);
}
function gn(t18) {
  return () => t18;
}
function yn(t18, e2) {
  return (n2) => t18 + n2 * e2;
}
function mn(t18) {
  return (t18 = +t18) == 1 ? _n : (e2, n2) => n2 - e2 ? function(t19, e3, n3) {
    return t19 = Math.pow(t19, n3), e3 = Math.pow(e3, n3) - t19, n3 = 1 / n3, (r2) => Math.pow(t19 + r2 * e3, n3);
  }(e2, n2, t18) : gn(isNaN(e2) ? n2 : e2);
}
function _n(t18, e2) {
  var n2 = e2 - t18;
  return n2 ? yn(t18, n2) : gn(isNaN(t18) ? e2 : t18);
}
ze(Ve, en, { copy: function(t18) {
  return Object.assign(new this.constructor(), this, t18);
}, displayable: function() {
  return this.rgb().displayable();
}, hex: Ke, formatHex: Ke, formatHsl: function() {
  return fn(this).formatHsl();
}, formatRgb: tn, toString: tn }), ze(un, on, qe(Ve, { brighter: function(t18) {
  return t18 = t18 == null ? Ue : Math.pow(Ue, t18), new un(this.r * t18, this.g * t18, this.b * t18, this.opacity);
}, darker: function(t18) {
  return t18 = t18 == null ? 0.7 : Math.pow(0.7, t18), new un(this.r * t18, this.g * t18, this.b * t18, this.opacity);
}, rgb: function() {
  return this;
}, displayable: function() {
  return -0.5 <= this.r && this.r < 255.5 && -0.5 <= this.g && this.g < 255.5 && -0.5 <= this.b && this.b < 255.5 && 0 <= this.opacity && this.opacity <= 1;
}, hex: sn, formatHex: sn, formatRgb: ln, toString: ln })), ze(dn, pn, qe(Ve, { brighter: function(t18) {
  return t18 = t18 == null ? Ue : Math.pow(Ue, t18), new dn(this.h, this.s, this.l * t18, this.opacity);
}, darker: function(t18) {
  return t18 = t18 == null ? 0.7 : Math.pow(0.7, t18), new dn(this.h, this.s, this.l * t18, this.opacity);
}, rgb: function() {
  var t18 = this.h % 360 + 360 * (this.h < 0), e2 = isNaN(t18) || isNaN(this.s) ? 0 : this.s, n2 = this.l, r2 = n2 + (n2 < 0.5 ? n2 : 1 - n2) * e2, i2 = 2 * n2 - r2;
  return new un(vn(t18 >= 240 ? t18 - 240 : t18 + 120, i2, r2), vn(t18, i2, r2), vn(t18 < 120 ? t18 + 240 : t18 - 120, i2, r2), this.opacity);
}, displayable: function() {
  return (0 <= this.s && this.s <= 1 || isNaN(this.s)) && 0 <= this.l && this.l <= 1 && 0 <= this.opacity && this.opacity <= 1;
}, formatHsl: function() {
  var t18 = this.opacity;
  return ((t18 = isNaN(t18) ? 1 : Math.max(0, Math.min(1, t18))) === 1 ? "hsl(" : "hsla(") + (this.h || 0) + ", " + 100 * (this.s || 0) + "%, " + 100 * (this.l || 0) + "%" + (t18 === 1 ? ")" : ", " + t18 + ")");
} }));
var bn = function t2(e2) {
  var n2 = mn(e2);
  function r2(t18, e3) {
    var r3 = n2((t18 = on(t18)).r, (e3 = on(e3)).r), i2 = n2(t18.g, e3.g), a2 = n2(t18.b, e3.b), o2 = _n(t18.opacity, e3.opacity);
    return (e4) => (t18.r = r3(e4), t18.g = i2(e4), t18.b = a2(e4), t18.opacity = o2(e4), t18 + "");
  }
  return r2.gamma = t2, r2;
}(1);
function wn(t18, e2) {
  e2 || (e2 = []);
  var n2, r2 = t18 ? Math.min(e2.length, t18.length) : 0, i2 = e2.slice();
  return (a2) => {
    for (n2 = 0; n2 < r2; ++n2)
      i2[n2] = t18[n2] * (1 - a2) + e2[n2] * a2;
    return i2;
  };
}
function xn(t18, e2) {
  var n2, r2 = e2 ? e2.length : 0, i2 = t18 ? Math.min(r2, t18.length) : 0, a2 = new Array(i2), o2 = new Array(r2);
  for (n2 = 0; n2 < i2; ++n2)
    a2[n2] = kn(t18[n2], e2[n2]);
  for (; n2 < r2; ++n2)
    o2[n2] = e2[n2];
  return (t19) => {
    for (n2 = 0; n2 < i2; ++n2)
      o2[n2] = a2[n2](t19);
    return o2;
  };
}
function Mn(t18, e2) {
  var n2 = new Date();
  return t18 = +t18, e2 = +e2, (r2) => (n2.setTime(t18 * (1 - r2) + e2 * r2), n2);
}
function Sn(t18, e2) {
  return t18 = +t18, e2 = +e2, (n2) => t18 * (1 - n2) + e2 * n2;
}
function An(t18, e2) {
  var n2, r2 = {}, i2 = {};
  for (n2 in t18 !== null && typeof t18 == "object" || (t18 = {}), e2 !== null && typeof e2 == "object" || (e2 = {}), e2)
    n2 in t18 ? r2[n2] = kn(t18[n2], e2[n2]) : i2[n2] = e2[n2];
  return (t19) => {
    for (n2 in r2)
      i2[n2] = r2[n2](t19);
    return i2;
  };
}
var Nn = /[-+]?(?:\d+\.?\d*|\.?\d+)(?:[eE][-+]?\d+)?/g, Cn = new RegExp(Nn.source, "g");
function On(t18, e2) {
  var n2, r2, i2, a2 = Nn.lastIndex = Cn.lastIndex = 0, o2 = -1, u2 = [], s2 = [];
  for (t18 += "", e2 += ""; (n2 = Nn.exec(t18)) && (r2 = Cn.exec(e2)); )
    (i2 = r2.index) > a2 && (i2 = e2.slice(a2, i2), u2[o2] ? u2[o2] += i2 : u2[++o2] = i2), (n2 = n2[0]) === (r2 = r2[0]) ? u2[o2] ? u2[o2] += r2 : u2[++o2] = r2 : (u2[++o2] = null, s2.push({ i: o2, x: Sn(n2, r2) })), a2 = Cn.lastIndex;
  return a2 < e2.length && (i2 = e2.slice(a2), u2[o2] ? u2[o2] += i2 : u2[++o2] = i2), u2.length < 2 ? s2[0] ? function(t19) {
    return (e3) => t19(e3) + "";
  }(s2[0].x) : function(t19) {
    return () => t19;
  }(e2) : (e2 = s2.length, (t19) => {
    for (var n3, r3 = 0; r3 < e2; ++r3)
      u2[(n3 = s2[r3]).i] = n3.x(t19);
    return u2.join("");
  });
}
function kn(t18, e2) {
  var n2, r2 = typeof e2;
  return e2 == null || r2 === "boolean" ? gn(e2) : (r2 === "number" ? Sn : r2 === "string" ? (n2 = en(e2)) ? (e2 = n2, bn) : On : e2 instanceof en ? bn : e2 instanceof Date ? Mn : function(t19) {
    return ArrayBuffer.isView(t19) && !(t19 instanceof DataView);
  }(e2) ? wn : Array.isArray(e2) ? xn : typeof e2.valueOf != "function" && typeof e2.toString != "function" || isNaN(e2) ? An : Sn)(t18, e2);
}
function jn(t18, e2) {
  return t18 = +t18, e2 = +e2, (n2) => Math.round(t18 * (1 - n2) + e2 * n2);
}
var En, In, Tn, Ln, Pn = 180 / Math.PI, Rn = { translateX: 0, translateY: 0, rotate: 0, skewX: 0, scaleX: 1, scaleY: 1 };
function $n(t18, e2, n2, r2, i2, a2) {
  var o2, u2, s2;
  return (o2 = Math.sqrt(t18 * t18 + e2 * e2)) && (t18 /= o2, e2 /= o2), (s2 = t18 * n2 + e2 * r2) && (n2 -= t18 * s2, r2 -= e2 * s2), (u2 = Math.sqrt(n2 * n2 + r2 * r2)) && (n2 /= u2, r2 /= u2, s2 /= u2), t18 * r2 < e2 * n2 && (t18 = -t18, e2 = -e2, s2 = -s2, o2 = -o2), { translateX: i2, translateY: a2, rotate: Math.atan2(e2, t18) * Pn, skewX: Math.atan(s2) * Pn, scaleX: o2, scaleY: u2 };
}
function zn(t18, e2, n2, r2) {
  function i2(t19) {
    return t19.length ? t19.pop() + " " : "";
  }
  return (a2, o2) => {
    var u2 = [], s2 = [];
    return a2 = t18(a2), o2 = t18(o2), function(t19, r3, i3, a3, o3, u3) {
      if (t19 !== i3 || r3 !== a3) {
        var s3 = o3.push("translate(", null, e2, null, n2);
        u3.push({ i: s3 - 4, x: Sn(t19, i3) }, { i: s3 - 2, x: Sn(r3, a3) });
      } else
        (i3 || a3) && o3.push("translate(" + i3 + e2 + a3 + n2);
    }(a2.translateX, a2.translateY, o2.translateX, o2.translateY, u2, s2), function(t19, e3, n3, a3) {
      t19 !== e3 ? (t19 - e3 > 180 ? e3 += 360 : e3 - t19 > 180 && (t19 += 360), a3.push({ i: n3.push(i2(n3) + "rotate(", null, r2) - 2, x: Sn(t19, e3) })) : e3 && n3.push(i2(n3) + "rotate(" + e3 + r2);
    }(a2.rotate, o2.rotate, u2, s2), function(t19, e3, n3, a3) {
      t19 !== e3 ? a3.push({ i: n3.push(i2(n3) + "skewX(", null, r2) - 2, x: Sn(t19, e3) }) : e3 && n3.push(i2(n3) + "skewX(" + e3 + r2);
    }(a2.skewX, o2.skewX, u2, s2), function(t19, e3, n3, r3, a3, o3) {
      if (t19 !== n3 || e3 !== r3) {
        var u3 = a3.push(i2(a3) + "scale(", null, ",", null, ")");
        o3.push({ i: u3 - 4, x: Sn(t19, n3) }, { i: u3 - 2, x: Sn(e3, r3) });
      } else
        n3 === 1 && r3 === 1 || a3.push(i2(a3) + "scale(" + n3 + "," + r3 + ")");
    }(a2.scaleX, a2.scaleY, o2.scaleX, o2.scaleY, u2, s2), a2 = o2 = null, (t19) => {
      for (var e3, n3 = -1, r3 = s2.length; ++n3 < r3; )
        u2[(e3 = s2[n3]).i] = e3.x(t19);
      return u2.join("");
    };
  };
}
var qn = zn(function(t18) {
  return t18 === "none" ? Rn : (En || (En = document.createElement("DIV"), In = document.documentElement, Tn = document.defaultView), En.style.transform = t18, t18 = Tn.getComputedStyle(In.appendChild(En), null).getPropertyValue("transform"), In.removeChild(En), $n(+(t18 = t18.slice(7, -1).split(","))[0], +t18[1], +t18[2], +t18[3], +t18[4], +t18[5]));
}, "px, ", "px)", "deg)"), Vn = zn(function(t18) {
  return t18 == null ? Rn : (Ln || (Ln = document.createElementNS("http://www.w3.org/2000/svg", "g")), Ln.setAttribute("transform", t18), (t18 = Ln.transform.baseVal.consolidate()) ? $n((t18 = t18.matrix).a, t18.b, t18.c, t18.d, t18.e, t18.f) : Rn);
}, ", ", ")", ")");
var Un, Fn, Bn = function(t18) {
  return (e2, n2) => {
    var r2 = t18((e2 = pn(e2)).h, (n2 = pn(n2)).h), i2 = _n(e2.s, n2.s), a2 = _n(e2.l, n2.l), o2 = _n(e2.opacity, n2.opacity);
    return (t19) => (e2.h = r2(t19), e2.s = i2(t19), e2.l = a2(t19), e2.opacity = o2(t19), e2 + "");
  };
}(function(t18, e2) {
  var n2 = e2 - t18;
  return n2 ? yn(t18, n2 > 180 || n2 < -180 ? n2 - 360 * Math.round(n2 / 360) : n2) : gn(isNaN(t18) ? e2 : t18);
}), Dn = 0, Wn = 0, Hn = 0, Xn = 0, Yn = 0, Qn = 0, Gn = typeof performance == "object" && performance.now ? performance : Date, Zn = typeof window == "object" && window.requestAnimationFrame ? window.requestAnimationFrame.bind(window) : (t18) => {
  setTimeout(t18, 17);
};
function Jn() {
  return Yn || (Zn(Kn), Yn = Gn.now() + Qn);
}
function Kn() {
  Yn = 0;
}
function tr() {
  this._call = this._time = this._next = null;
}
function er(t18, e2, n2) {
  var r2 = new tr();
  return r2.restart(t18, e2, n2), r2;
}
function nr() {
  Yn = (Xn = Gn.now()) + Qn, Dn = Wn = 0;
  try {
    !function() {
      Jn(), ++Dn;
      for (var t18, e2 = Un; e2; )
        (t18 = Yn - e2._time) >= 0 && e2._call.call(null, t18), e2 = e2._next;
      --Dn;
    }();
  } finally {
    Dn = 0, function() {
      var t18, e2, n2 = Un, r2 = 1 / 0;
      for (; n2; )
        n2._call ? (r2 > n2._time && (r2 = n2._time), t18 = n2, n2 = n2._next) : (e2 = n2._next, n2._next = null, n2 = t18 ? t18._next = e2 : Un = e2);
      Fn = t18, ir(r2);
    }(), Yn = 0;
  }
}
function rr() {
  var t18 = Gn.now(), e2 = t18 - Xn;
  e2 > 1e3 && (Qn -= e2, Xn = t18);
}
function ir(t18) {
  Dn || (Wn && (Wn = clearTimeout(Wn)), t18 - Yn > 24 ? (t18 < 1 / 0 && (Wn = setTimeout(nr, t18 - Gn.now() - Qn)), Hn && (Hn = clearInterval(Hn))) : (Hn || (Xn = Gn.now(), Hn = setInterval(rr, 1e3)), Dn = 1, Zn(nr)));
}
function ar(t18, e2, n2) {
  var r2 = new tr();
  return e2 = e2 == null ? 0 : +e2, r2.restart((n3) => {
    r2.stop(), t18(n3 + e2);
  }, e2, n2), r2;
}
tr.prototype = er.prototype = { constructor: tr, restart: function(t18, e2, n2) {
  if (typeof t18 != "function")
    throw new TypeError("callback is not a function");
  n2 = (n2 == null ? Jn() : +n2) + (e2 == null ? 0 : +e2), this._next || Fn === this || (Fn ? Fn._next = this : Un = this, Fn = this), this._call = t18, this._time = n2, ir();
}, stop: function() {
  this._call && (this._call = null, this._time = 1 / 0, ir());
} };
var or = St("start", "end", "cancel", "interrupt"), ur = [];
function sr(t18, e2, n2, r2, i2, a2) {
  var o2 = t18.__transition;
  if (o2) {
    if (n2 in o2)
      return;
  } else
    t18.__transition = {};
  !function(t19, e3, n3) {
    var r3, i3 = t19.__transition;
    function a3(t20) {
      n3.state = 1, n3.timer.restart(o3, n3.delay, n3.time), n3.delay <= t20 && o3(t20 - n3.delay);
    }
    function o3(a4) {
      var l2, c2, h2, f2;
      if (n3.state !== 1)
        return s2();
      for (l2 in i3)
        if ((f2 = i3[l2]).name === n3.name) {
          if (f2.state === 3)
            return ar(o3);
          f2.state === 4 ? (f2.state = 6, f2.timer.stop(), f2.on.call("interrupt", t19, t19.__data__, f2.index, f2.group), delete i3[l2]) : +l2 < e3 && (f2.state = 6, f2.timer.stop(), f2.on.call("cancel", t19, t19.__data__, f2.index, f2.group), delete i3[l2]);
        }
      if (ar(() => {
        n3.state === 3 && (n3.state = 4, n3.timer.restart(u2, n3.delay, n3.time), u2(a4));
      }), n3.state = 2, n3.on.call("start", t19, t19.__data__, n3.index, n3.group), n3.state === 2) {
        for (n3.state = 3, r3 = new Array(h2 = n3.tween.length), l2 = 0, c2 = -1; l2 < h2; ++l2)
          (f2 = n3.tween[l2].value.call(t19, t19.__data__, n3.index, n3.group)) && (r3[++c2] = f2);
        r3.length = c2 + 1;
      }
    }
    function u2(e4) {
      for (var i4 = e4 < n3.duration ? n3.ease.call(null, e4 / n3.duration) : (n3.timer.restart(s2), n3.state = 5, 1), a4 = -1, o4 = r3.length; ++a4 < o4; )
        r3[a4].call(t19, i4);
      n3.state === 5 && (n3.on.call("end", t19, t19.__data__, n3.index, n3.group), s2());
    }
    function s2() {
      for (var r4 in n3.state = 6, n3.timer.stop(), delete i3[e3], i3)
        return;
      delete t19.__transition;
    }
    i3[e3] = n3, n3.timer = er(a3, 0, n3.time);
  }(t18, n2, { name: e2, index: r2, group: i2, on: or, tween: ur, time: a2.time, delay: a2.delay, duration: a2.duration, ease: a2.ease, timer: null, state: 0 });
}
function lr(t18, e2) {
  var n2 = hr(t18, e2);
  if (n2.state > 0)
    throw new Error("too late; already scheduled");
  return n2;
}
function cr(t18, e2) {
  var n2 = hr(t18, e2);
  if (n2.state > 3)
    throw new Error("too late; already running");
  return n2;
}
function hr(t18, e2) {
  var n2 = t18.__transition;
  if (!n2 || !(n2 = n2[e2]))
    throw new Error("transition not found");
  return n2;
}
function fr(t18, e2) {
  var n2, r2;
  return function() {
    var i2 = cr(this, t18), a2 = i2.tween;
    if (a2 !== n2) {
      for (var o2 = 0, u2 = (r2 = n2 = a2).length; o2 < u2; ++o2)
        if (r2[o2].name === e2) {
          (r2 = r2.slice()).splice(o2, 1);
          break;
        }
    }
    i2.tween = r2;
  };
}
function pr(t18, e2, n2) {
  var r2, i2;
  if (typeof n2 != "function")
    throw new Error();
  return function() {
    var a2 = cr(this, t18), o2 = a2.tween;
    if (o2 !== r2) {
      i2 = (r2 = o2).slice();
      for (var u2 = { name: e2, value: n2 }, s2 = 0, l2 = i2.length; s2 < l2; ++s2)
        if (i2[s2].name === e2) {
          i2[s2] = u2;
          break;
        }
      s2 === l2 && i2.push(u2);
    }
    a2.tween = i2;
  };
}
function dr(t18, e2, n2) {
  var r2 = t18._id;
  return t18.each(function() {
    var t19 = cr(this, r2);
    (t19.value || (t19.value = {}))[e2] = n2.apply(this, arguments);
  }), (t19) => hr(t19, r2).value[e2];
}
function vr(t18, e2) {
  var n2;
  return (typeof e2 == "number" ? Sn : e2 instanceof en ? bn : (n2 = en(e2)) ? (e2 = n2, bn) : On)(t18, e2);
}
function gr(t18) {
  return function() {
    this.removeAttribute(t18);
  };
}
function yr(t18) {
  return function() {
    this.removeAttributeNS(t18.space, t18.local);
  };
}
function mr(t18, e2, n2) {
  var r2, i2, a2 = n2 + "";
  return function() {
    var o2 = this.getAttribute(t18);
    return o2 === a2 ? null : o2 === r2 ? i2 : i2 = e2(r2 = o2, n2);
  };
}
function _r(t18, e2, n2) {
  var r2, i2, a2 = n2 + "";
  return function() {
    var o2 = this.getAttributeNS(t18.space, t18.local);
    return o2 === a2 ? null : o2 === r2 ? i2 : i2 = e2(r2 = o2, n2);
  };
}
function br(t18, e2, n2) {
  var r2, i2, a2;
  return function() {
    var o2, u2, s2 = n2(this);
    if (s2 != null)
      return (o2 = this.getAttribute(t18)) === (u2 = s2 + "") ? null : o2 === r2 && u2 === i2 ? a2 : (i2 = u2, a2 = e2(r2 = o2, s2));
    this.removeAttribute(t18);
  };
}
function wr(t18, e2, n2) {
  var r2, i2, a2;
  return function() {
    var o2, u2, s2 = n2(this);
    if (s2 != null)
      return (o2 = this.getAttributeNS(t18.space, t18.local)) === (u2 = s2 + "") ? null : o2 === r2 && u2 === i2 ? a2 : (i2 = u2, a2 = e2(r2 = o2, s2));
    this.removeAttributeNS(t18.space, t18.local);
  };
}
function xr(t18, e2) {
  return function(n2) {
    this.setAttribute(t18, e2.call(this, n2));
  };
}
function Mr(t18, e2) {
  return function(n2) {
    this.setAttributeNS(t18.space, t18.local, e2.call(this, n2));
  };
}
function Sr(t18, e2) {
  var n2, r2;
  function i2() {
    var i3 = e2.apply(this, arguments);
    return i3 !== r2 && (n2 = (r2 = i3) && Mr(t18, i3)), n2;
  }
  return i2._value = e2, i2;
}
function Ar(t18, e2) {
  var n2, r2;
  function i2() {
    var i3 = e2.apply(this, arguments);
    return i3 !== r2 && (n2 = (r2 = i3) && xr(t18, i3)), n2;
  }
  return i2._value = e2, i2;
}
function Nr(t18, e2) {
  return function() {
    lr(this, t18).delay = +e2.apply(this, arguments);
  };
}
function Cr(t18, e2) {
  return e2 = +e2, function() {
    lr(this, t18).delay = e2;
  };
}
function Or(t18, e2) {
  return function() {
    cr(this, t18).duration = +e2.apply(this, arguments);
  };
}
function kr(t18, e2) {
  return e2 = +e2, function() {
    cr(this, t18).duration = e2;
  };
}
function jr(t18, e2) {
  if (typeof e2 != "function")
    throw new Error();
  return function() {
    cr(this, t18).ease = e2;
  };
}
function Er(t18, e2, n2) {
  var r2, i2, a2 = function(t19) {
    return (t19 + "").trim().split(/^|\s+/).every((t20) => {
      var e3 = t20.indexOf(".");
      return e3 >= 0 && (t20 = t20.slice(0, e3)), !t20 || t20 === "start";
    });
  }(e2) ? lr : cr;
  return function() {
    var o2 = a2(this, t18), u2 = o2.on;
    u2 !== r2 && (i2 = (r2 = u2).copy()).on(e2, n2), o2.on = i2;
  };
}
var Ir = Re.prototype.constructor;
function Tr(t18) {
  return function() {
    this.style.removeProperty(t18);
  };
}
function Lr(t18, e2, n2) {
  return function(r2) {
    this.style.setProperty(t18, e2.call(this, r2), n2);
  };
}
function Pr(t18, e2, n2) {
  var r2, i2;
  function a2() {
    var a3 = e2.apply(this, arguments);
    return a3 !== i2 && (r2 = (i2 = a3) && Lr(t18, a3, n2)), r2;
  }
  return a2._value = e2, a2;
}
function Rr(t18) {
  return function(e2) {
    this.textContent = t18.call(this, e2);
  };
}
function $r(t18) {
  var e2, n2;
  function r2() {
    var r3 = t18.apply(this, arguments);
    return r3 !== n2 && (e2 = (n2 = r3) && Rr(r3)), e2;
  }
  return r2._value = t18, r2;
}
var zr = 0;
function qr(t18, e2, n2, r2) {
  this._groups = t18, this._parents = e2, this._name = n2, this._id = r2;
}
function Vr() {
  return ++zr;
}
var Ur = Re.prototype;
function Fr(t18) {
  return +t18;
}
function Br(t18) {
  return t18 * t18;
}
function Dr(t18) {
  return t18 * (2 - t18);
}
function Wr(t18) {
  return ((t18 *= 2) <= 1 ? t18 * t18 : --t18 * (2 - t18) + 1) / 2;
}
function Hr(t18) {
  return t18 * t18 * t18;
}
function Xr(t18) {
  return --t18 * t18 * t18 + 1;
}
function Yr(t18) {
  return ((t18 *= 2) <= 1 ? t18 * t18 * t18 : (t18 -= 2) * t18 * t18 + 2) / 2;
}
qr.prototype = { constructor: qr, select: function(t18) {
  var e2 = this._name, n2 = this._id;
  typeof t18 != "function" && (t18 = Rt(t18));
  for (var r2 = this._groups, i2 = r2.length, a2 = new Array(i2), o2 = 0; o2 < i2; ++o2)
    for (var u2, s2, l2 = r2[o2], c2 = l2.length, h2 = a2[o2] = new Array(c2), f2 = 0; f2 < c2; ++f2)
      (u2 = l2[f2]) && (s2 = t18.call(u2, u2.__data__, f2, l2)) && ("__data__" in u2 && (s2.__data__ = u2.__data__), h2[f2] = s2, sr(h2[f2], e2, n2, f2, h2, hr(u2, n2)));
  return new qr(a2, this._parents, e2, n2);
}, selectAll: function(t18) {
  var e2 = this._name, n2 = this._id;
  typeof t18 != "function" && (t18 = zt(t18));
  for (var r2 = this._groups, i2 = r2.length, a2 = [], o2 = [], u2 = 0; u2 < i2; ++u2)
    for (var s2, l2 = r2[u2], c2 = l2.length, h2 = 0; h2 < c2; ++h2)
      if (s2 = l2[h2]) {
        for (var f2, p2 = t18.call(s2, s2.__data__, h2, l2), d2 = hr(s2, n2), v2 = 0, g2 = p2.length; v2 < g2; ++v2)
          (f2 = p2[v2]) && sr(f2, e2, n2, v2, p2, d2);
        a2.push(p2), o2.push(s2);
      }
  return new qr(a2, o2, e2, n2);
}, filter: function(t18) {
  typeof t18 != "function" && (t18 = qt(t18));
  for (var e2 = this._groups, n2 = e2.length, r2 = new Array(n2), i2 = 0; i2 < n2; ++i2)
    for (var a2, o2 = e2[i2], u2 = o2.length, s2 = r2[i2] = [], l2 = 0; l2 < u2; ++l2)
      (a2 = o2[l2]) && t18.call(a2, a2.__data__, l2, o2) && s2.push(a2);
  return new qr(r2, this._parents, this._name, this._id);
}, merge: function(t18) {
  if (t18._id !== this._id)
    throw new Error();
  for (var e2 = this._groups, n2 = t18._groups, r2 = e2.length, i2 = n2.length, a2 = Math.min(r2, i2), o2 = new Array(r2), u2 = 0; u2 < a2; ++u2)
    for (var s2, l2 = e2[u2], c2 = n2[u2], h2 = l2.length, f2 = o2[u2] = new Array(h2), p2 = 0; p2 < h2; ++p2)
      (s2 = l2[p2] || c2[p2]) && (f2[p2] = s2);
  for (; u2 < r2; ++u2)
    o2[u2] = e2[u2];
  return new qr(o2, this._parents, this._name, this._id);
}, selection: function() {
  return new Ir(this._groups, this._parents);
}, transition: function() {
  for (var t18 = this._name, e2 = this._id, n2 = Vr(), r2 = this._groups, i2 = r2.length, a2 = 0; a2 < i2; ++a2)
    for (var o2, u2 = r2[a2], s2 = u2.length, l2 = 0; l2 < s2; ++l2)
      if (o2 = u2[l2]) {
        var c2 = hr(o2, e2);
        sr(o2, t18, n2, l2, u2, { time: c2.time + c2.delay + c2.duration, delay: 0, duration: c2.duration, ease: c2.ease });
      }
  return new qr(r2, this._parents, t18, n2);
}, call: Ur.call, nodes: Ur.nodes, node: Ur.node, size: Ur.size, empty: Ur.empty, each: Ur.each, on: function(t18, e2) {
  var n2 = this._id;
  return arguments.length < 2 ? hr(this.node(), n2).on.on(t18) : this.each(Er(n2, t18, e2));
}, attr: function(t18, e2) {
  var n2 = Et(t18), r2 = n2 === "transform" ? Vn : vr;
  return this.attrTween(t18, typeof e2 == "function" ? (n2.local ? wr : br)(n2, r2, dr(this, "attr." + t18, e2)) : e2 == null ? (n2.local ? yr : gr)(n2) : (n2.local ? _r : mr)(n2, r2, e2));
}, attrTween: function(t18, e2) {
  var n2 = "attr." + t18;
  if (arguments.length < 2)
    return (n2 = this.tween(n2)) && n2._value;
  if (e2 == null)
    return this.tween(n2, null);
  if (typeof e2 != "function")
    throw new Error();
  var r2 = Et(t18);
  return this.tween(n2, (r2.local ? Sr : Ar)(r2, e2));
}, style: function(t18, e2, n2) {
  var r2 = (t18 += "") == "transform" ? qn : vr;
  return e2 == null ? this.styleTween(t18, function(t19, e3) {
    var n3, r3, i2;
    return function() {
      var a2 = ee(this, t19), o2 = (this.style.removeProperty(t19), ee(this, t19));
      return a2 === o2 ? null : a2 === n3 && o2 === r3 ? i2 : i2 = e3(n3 = a2, r3 = o2);
    };
  }(t18, r2)).on("end.style." + t18, Tr(t18)) : typeof e2 == "function" ? this.styleTween(t18, function(t19, e3, n3) {
    var r3, i2, a2;
    return function() {
      var o2 = ee(this, t19), u2 = n3(this), s2 = u2 + "";
      return u2 == null && (this.style.removeProperty(t19), s2 = u2 = ee(this, t19)), o2 === s2 ? null : o2 === r3 && s2 === i2 ? a2 : (i2 = s2, a2 = e3(r3 = o2, u2));
    };
  }(t18, r2, dr(this, "style." + t18, e2))).each(function(t19, e3) {
    var n3, r3, i2, a2, o2 = "style." + e3, u2 = "end." + o2;
    return function() {
      var s2 = cr(this, t19), l2 = s2.on, c2 = s2.value[o2] == null ? a2 || (a2 = Tr(e3)) : void 0;
      l2 === n3 && i2 === c2 || (r3 = (n3 = l2).copy()).on(u2, i2 = c2), s2.on = r3;
    };
  }(this._id, t18)) : this.styleTween(t18, function(t19, e3, n3) {
    var r3, i2, a2 = n3 + "";
    return function() {
      var o2 = ee(this, t19);
      return o2 === a2 ? null : o2 === r3 ? i2 : i2 = e3(r3 = o2, n3);
    };
  }(t18, r2, e2), n2).on("end.style." + t18, null);
}, styleTween: function(t18, e2, n2) {
  var r2 = "style." + (t18 += "");
  if (arguments.length < 2)
    return (r2 = this.tween(r2)) && r2._value;
  if (e2 == null)
    return this.tween(r2, null);
  if (typeof e2 != "function")
    throw new Error();
  return this.tween(r2, Pr(t18, e2, n2 != null ? n2 : ""));
}, text: function(t18) {
  return this.tween("text", typeof t18 == "function" ? function(t19) {
    return function() {
      var e2 = t19(this);
      this.textContent = e2 != null ? e2 : "";
    };
  }(dr(this, "text", t18)) : function(t19) {
    return function() {
      this.textContent = t19;
    };
  }(t18 == null ? "" : t18 + ""));
}, textTween: function(t18) {
  var e2 = "text";
  if (arguments.length < 1)
    return (e2 = this.tween(e2)) && e2._value;
  if (t18 == null)
    return this.tween(e2, null);
  if (typeof t18 != "function")
    throw new Error();
  return this.tween(e2, $r(t18));
}, remove: function() {
  return this.on("end.remove", function(t18) {
    return function() {
      var e2 = this.parentNode;
      for (var n2 in this.__transition)
        if (+n2 !== t18)
          return;
      e2 && e2.removeChild(this);
    };
  }(this._id));
}, tween: function(t18, e2) {
  var n2 = this._id;
  if (t18 += "", arguments.length < 2) {
    for (var r2, i2 = hr(this.node(), n2).tween, a2 = 0, o2 = i2.length; a2 < o2; ++a2)
      if ((r2 = i2[a2]).name === t18)
        return r2.value;
    return null;
  }
  return this.each((e2 == null ? fr : pr)(n2, t18, e2));
}, delay: function(t18) {
  var e2 = this._id;
  return arguments.length ? this.each((typeof t18 == "function" ? Nr : Cr)(e2, t18)) : hr(this.node(), e2).delay;
}, duration: function(t18) {
  var e2 = this._id;
  return arguments.length ? this.each((typeof t18 == "function" ? Or : kr)(e2, t18)) : hr(this.node(), e2).duration;
}, ease: function(t18) {
  var e2 = this._id;
  return arguments.length ? this.each(jr(e2, t18)) : hr(this.node(), e2).ease;
}, end: function() {
  var t18, e2, n2 = this, r2 = n2._id, i2 = n2.size();
  return new Promise(function(a2, o2) {
    var u2 = { value: o2 }, s2 = { value: () => {
      --i2 == 0 && a2();
    } };
    n2.each(function() {
      var n3 = cr(this, r2), i3 = n3.on;
      i3 !== t18 && ((e2 = (t18 = i3).copy())._.cancel.push(u2), e2._.interrupt.push(u2), e2._.end.push(s2)), n3.on = e2;
    });
  });
} };
var Qr = function t3(e2) {
  function n2(t18) {
    return Math.pow(t18, e2);
  }
  return e2 = +e2, n2.exponent = t3, n2;
}(3), Gr = function t4(e2) {
  function n2(t18) {
    return 1 - Math.pow(1 - t18, e2);
  }
  return e2 = +e2, n2.exponent = t4, n2;
}(3), Zr = function t5(e2) {
  function n2(t18) {
    return ((t18 *= 2) <= 1 ? Math.pow(t18, e2) : 2 - Math.pow(2 - t18, e2)) / 2;
  }
  return e2 = +e2, n2.exponent = t5, n2;
}(3), Jr = Math.PI, Kr = Jr / 2;
function ti(t18) {
  return +t18 == 1 ? 1 : 1 - Math.cos(t18 * Kr);
}
function ei(t18) {
  return Math.sin(t18 * Kr);
}
function ni(t18) {
  return (1 - Math.cos(Jr * t18)) / 2;
}
function ri(t18) {
  return 1.0009775171065494 * (Math.pow(2, -10 * t18) - 9765625e-10);
}
function ii(t18) {
  return ri(1 - +t18);
}
function ai(t18) {
  return 1 - ri(t18);
}
function oi(t18) {
  return ((t18 *= 2) <= 1 ? ri(1 - t18) : 2 - ri(t18 - 1)) / 2;
}
function ui(t18) {
  return 1 - Math.sqrt(1 - t18 * t18);
}
function si(t18) {
  return Math.sqrt(1 - --t18 * t18);
}
function li(t18) {
  return ((t18 *= 2) <= 1 ? 1 - Math.sqrt(1 - t18 * t18) : Math.sqrt(1 - (t18 -= 2) * t18) + 1) / 2;
}
var ci = 4 / 11, hi = 7.5625;
function fi(t18) {
  return 1 - pi(1 - t18);
}
function pi(t18) {
  return (t18 = +t18) < ci ? hi * t18 * t18 : t18 < 0.7272727272727273 ? hi * (t18 -= 0.5454545454545454) * t18 + 0.75 : t18 < 0.9090909090909091 ? hi * (t18 -= 0.8181818181818182) * t18 + 0.9375 : hi * (t18 -= 0.9545454545454546) * t18 + 0.984375;
}
function di(t18) {
  return ((t18 *= 2) <= 1 ? 1 - pi(1 - t18) : pi(t18 - 1) + 1) / 2;
}
var vi = function t6(e2) {
  function n2(t18) {
    return (t18 = +t18) * t18 * (e2 * (t18 - 1) + t18);
  }
  return e2 = +e2, n2.overshoot = t6, n2;
}(1.70158), gi = function t7(e2) {
  function n2(t18) {
    return --t18 * t18 * ((t18 + 1) * e2 + t18) + 1;
  }
  return e2 = +e2, n2.overshoot = t7, n2;
}(1.70158), yi = function t8(e2) {
  function n2(t18) {
    return ((t18 *= 2) < 1 ? t18 * t18 * ((e2 + 1) * t18 - e2) : (t18 -= 2) * t18 * ((e2 + 1) * t18 + e2) + 2) / 2;
  }
  return e2 = +e2, n2.overshoot = t8, n2;
}(1.70158), mi = 2 * Math.PI, _i = function t9(e2, n2) {
  var r2 = Math.asin(1 / (e2 = Math.max(1, e2))) * (n2 /= mi);
  function i2(t18) {
    return e2 * ri(- --t18) * Math.sin((r2 - t18) / n2);
  }
  return i2.amplitude = (e3) => t9(e3, n2 * mi), i2.period = (n3) => t9(e2, n3), i2;
}(1, 0.3), bi = function t10(e2, n2) {
  var r2 = Math.asin(1 / (e2 = Math.max(1, e2))) * (n2 /= mi);
  function i2(t18) {
    return 1 - e2 * ri(t18 = +t18) * Math.sin((t18 + r2) / n2);
  }
  return i2.amplitude = (e3) => t10(e3, n2 * mi), i2.period = (n3) => t10(e2, n3), i2;
}(1, 0.3), wi = function t11(e2, n2) {
  var r2 = Math.asin(1 / (e2 = Math.max(1, e2))) * (n2 /= mi);
  function i2(t18) {
    return ((t18 = 2 * t18 - 1) < 0 ? e2 * ri(-t18) * Math.sin((r2 - t18) / n2) : 2 - e2 * ri(t18) * Math.sin((r2 + t18) / n2)) / 2;
  }
  return i2.amplitude = (e3) => t11(e3, n2 * mi), i2.period = (n3) => t11(e2, n3), i2;
}(1, 0.3), xi = { time: null, delay: 0, duration: 250, ease: Yr };
function Mi(t18, e2) {
  for (var n2; !(n2 = t18.__transition) || !(n2 = n2[e2]); )
    if (!(t18 = t18.parentNode))
      return xi.time = Jn(), xi;
  return n2;
}
Re.prototype.interrupt = function(t18) {
  return this.each(function() {
    !function(t19, e2) {
      var n2, r2, i2, a2 = t19.__transition, o2 = true;
      if (a2) {
        for (i2 in e2 = e2 == null ? null : e2 + "", a2)
          (n2 = a2[i2]).name === e2 ? (r2 = n2.state > 2 && n2.state < 5, n2.state = 6, n2.timer.stop(), n2.on.call(r2 ? "interrupt" : "cancel", t19, t19.__data__, n2.index, n2.group), delete a2[i2]) : o2 = false;
        o2 && delete t19.__transition;
      }
    }(this, t18);
  });
}, Re.prototype.transition = function(t18) {
  var e2, n2;
  t18 instanceof qr ? (e2 = t18._id, t18 = t18._name) : (e2 = Vr(), (n2 = xi).time = Jn(), t18 = t18 == null ? null : t18 + "");
  for (var r2 = this._groups, i2 = r2.length, a2 = 0; a2 < i2; ++a2)
    for (var o2, u2 = r2[a2], s2 = u2.length, l2 = 0; l2 < s2; ++l2)
      (o2 = u2[l2]) && sr(o2, t18, e2, l2, u2, n2 || Mi(o2, e2));
  return new qr(r2, this._parents, t18, e2);
};
var Si = Math.PI, Ai = 2 * Si, Ni = Ai - 1e-6;
function Ci() {
  this._x0 = this._y0 = this._x1 = this._y1 = null, this._ = "";
}
function Oi() {
  return new Ci();
}
Ci.prototype = Oi.prototype = { constructor: Ci, moveTo: function(t18, e2) {
  this._ += "M" + (this._x0 = this._x1 = +t18) + "," + (this._y0 = this._y1 = +e2);
}, closePath: function() {
  this._x1 !== null && (this._x1 = this._x0, this._y1 = this._y0, this._ += "Z");
}, lineTo: function(t18, e2) {
  this._ += "L" + (this._x1 = +t18) + "," + (this._y1 = +e2);
}, quadraticCurveTo: function(t18, e2, n2, r2) {
  this._ += "Q" + +t18 + "," + +e2 + "," + (this._x1 = +n2) + "," + (this._y1 = +r2);
}, bezierCurveTo: function(t18, e2, n2, r2, i2, a2) {
  this._ += "C" + +t18 + "," + +e2 + "," + +n2 + "," + +r2 + "," + (this._x1 = +i2) + "," + (this._y1 = +a2);
}, arcTo: function(t18, e2, n2, r2, i2) {
  t18 = +t18, e2 = +e2, n2 = +n2, r2 = +r2, i2 = +i2;
  var a2 = this._x1, o2 = this._y1, u2 = n2 - t18, s2 = r2 - e2, l2 = a2 - t18, c2 = o2 - e2, h2 = l2 * l2 + c2 * c2;
  if (i2 < 0)
    throw new Error("negative radius: " + i2);
  if (this._x1 === null)
    this._ += "M" + (this._x1 = t18) + "," + (this._y1 = e2);
  else if (h2 > 1e-6)
    if (Math.abs(c2 * u2 - s2 * l2) > 1e-6 && i2) {
      var f2 = n2 - a2, p2 = r2 - o2, d2 = u2 * u2 + s2 * s2, v2 = f2 * f2 + p2 * p2, g2 = Math.sqrt(d2), y2 = Math.sqrt(h2), m2 = i2 * Math.tan((Si - Math.acos((d2 + h2 - v2) / (2 * g2 * y2))) / 2), _2 = m2 / y2, b2 = m2 / g2;
      Math.abs(_2 - 1) > 1e-6 && (this._ += "L" + (t18 + _2 * l2) + "," + (e2 + _2 * c2)), this._ += "A" + i2 + "," + i2 + ",0,0," + +(c2 * f2 > l2 * p2) + "," + (this._x1 = t18 + b2 * u2) + "," + (this._y1 = e2 + b2 * s2);
    } else
      this._ += "L" + (this._x1 = t18) + "," + (this._y1 = e2);
  else
    ;
}, arc: function(t18, e2, n2, r2, i2, a2) {
  t18 = +t18, e2 = +e2, a2 = !!a2;
  var o2 = (n2 = +n2) * Math.cos(r2), u2 = n2 * Math.sin(r2), s2 = t18 + o2, l2 = e2 + u2, c2 = 1 ^ a2, h2 = a2 ? r2 - i2 : i2 - r2;
  if (n2 < 0)
    throw new Error("negative radius: " + n2);
  this._x1 === null ? this._ += "M" + s2 + "," + l2 : (Math.abs(this._x1 - s2) > 1e-6 || Math.abs(this._y1 - l2) > 1e-6) && (this._ += "L" + s2 + "," + l2), n2 && (h2 < 0 && (h2 = h2 % Ai + Ai), h2 > Ni ? this._ += "A" + n2 + "," + n2 + ",0,1," + c2 + "," + (t18 - o2) + "," + (e2 - u2) + "A" + n2 + "," + n2 + ",0,1," + c2 + "," + (this._x1 = s2) + "," + (this._y1 = l2) : h2 > 1e-6 && (this._ += "A" + n2 + "," + n2 + ",0," + +(h2 >= Si) + "," + c2 + "," + (this._x1 = t18 + n2 * Math.cos(i2)) + "," + (this._y1 = e2 + n2 * Math.sin(i2))));
}, rect: function(t18, e2, n2, r2) {
  this._ += "M" + (this._x0 = this._x1 = +t18) + "," + (this._y0 = this._y1 = +e2) + "h" + +n2 + "v" + +r2 + "h" + -n2 + "Z";
}, toString: function() {
  return this._;
} };
function ki() {
}
function ji(t18, e2) {
  var n2 = new ki();
  if (t18 instanceof ki)
    t18.each((t19, e3) => {
      n2.set(e3, t19);
    });
  else if (Array.isArray(t18)) {
    var r2, i2 = -1, a2 = t18.length;
    if (e2 == null)
      for (; ++i2 < a2; )
        n2.set(i2, t18[i2]);
    else
      for (; ++i2 < a2; )
        n2.set(e2(r2 = t18[i2], i2, t18), r2);
  } else if (t18)
    for (var o2 in t18)
      n2.set(o2, t18[o2]);
  return n2;
}
function Ei() {
}
ki.prototype = ji.prototype = { constructor: ki, has: function(t18) {
  return "$" + t18 in this;
}, get: function(t18) {
  return this["$" + t18];
}, set: function(t18, e2) {
  return this["$" + t18] = e2, this;
}, remove: function(t18) {
  var e2 = "$" + t18;
  return e2 in this && delete this[e2];
}, clear: function() {
  for (var t18 in this)
    t18[0] === "$" && delete this[t18];
}, keys: function() {
  var t18 = [];
  for (var e2 in this)
    e2[0] === "$" && t18.push(e2.slice(1));
  return t18;
}, values: function() {
  var t18 = [];
  for (var e2 in this)
    e2[0] === "$" && t18.push(this[e2]);
  return t18;
}, entries: function() {
  var t18 = [];
  for (var e2 in this)
    e2[0] === "$" && t18.push({ key: e2.slice(1), value: this[e2] });
  return t18;
}, size: function() {
  var t18 = 0;
  for (var e2 in this)
    e2[0] === "$" && ++t18;
  return t18;
}, empty: function() {
  for (var t18 in this)
    if (t18[0] === "$")
      return false;
  return true;
}, each: function(t18) {
  for (var e2 in this)
    e2[0] === "$" && t18(this[e2], e2.slice(1), this);
} };
var Ii = ji.prototype;
function Ti(t18, e2) {
  if ((n2 = (t18 = e2 ? t18.toExponential(e2 - 1) : t18.toExponential()).indexOf("e")) < 0)
    return null;
  var n2, r2 = t18.slice(0, n2);
  return [r2.length > 1 ? r2[0] + r2.slice(2) : r2, +t18.slice(n2 + 1)];
}
function Li(t18) {
  return (t18 = Ti(Math.abs(t18))) ? t18[1] : NaN;
}
Ei.prototype = { constructor: Ei, has: Ii.has, add: function(t18) {
  return this["$" + (t18 += "")] = t18, this;
}, remove: Ii.remove, clear: Ii.clear, values: Ii.keys, size: Ii.size, empty: Ii.empty, each: Ii.each };
var Pi, Ri = /^(?:(.)?([<>=^]))?([+\-( ])?([$#])?(0)?(\d+)?(,)?(\.\d+)?(~)?([a-z%])?$/i;
function $i(t18) {
  if (!(e2 = Ri.exec(t18)))
    throw new Error("invalid format: " + t18);
  var e2;
  return new zi({ fill: e2[1], align: e2[2], sign: e2[3], symbol: e2[4], zero: e2[5], width: e2[6], comma: e2[7], precision: e2[8] && e2[8].slice(1), trim: e2[9], type: e2[10] });
}
function zi(t18) {
  this.fill = t18.fill === void 0 ? " " : t18.fill + "", this.align = t18.align === void 0 ? ">" : t18.align + "", this.sign = t18.sign === void 0 ? "-" : t18.sign + "", this.symbol = t18.symbol === void 0 ? "" : t18.symbol + "", this.zero = !!t18.zero, this.width = t18.width === void 0 ? void 0 : +t18.width, this.comma = !!t18.comma, this.precision = t18.precision === void 0 ? void 0 : +t18.precision, this.trim = !!t18.trim, this.type = t18.type === void 0 ? "" : t18.type + "";
}
function qi(t18, e2) {
  var n2 = Ti(t18, e2);
  if (!n2)
    return t18 + "";
  var r2 = n2[0], i2 = n2[1];
  return i2 < 0 ? "0." + new Array(-i2).join("0") + r2 : r2.length > i2 + 1 ? r2.slice(0, i2 + 1) + "." + r2.slice(i2 + 1) : r2 + new Array(i2 - r2.length + 2).join("0");
}
$i.prototype = zi.prototype, zi.prototype.toString = function() {
  return this.fill + this.align + this.sign + this.symbol + (this.zero ? "0" : "") + (this.width === void 0 ? "" : Math.max(1, 0 | this.width)) + (this.comma ? "," : "") + (this.precision === void 0 ? "" : "." + Math.max(0, 0 | this.precision)) + (this.trim ? "~" : "") + this.type;
};
var Vi = { "%": (t18, e2) => (100 * t18).toFixed(e2), b: (t18) => Math.round(t18).toString(2), c: (t18) => t18 + "", d: function(t18) {
  return Math.abs(t18 = Math.round(t18)) >= 1e21 ? t18.toLocaleString("en").replace(/,/g, "") : t18.toString(10);
}, e: (t18, e2) => t18.toExponential(e2), f: (t18, e2) => t18.toFixed(e2), g: (t18, e2) => t18.toPrecision(e2), o: (t18) => Math.round(t18).toString(8), p: (t18, e2) => qi(100 * t18, e2), r: qi, s: function(t18, e2) {
  var n2 = Ti(t18, e2);
  if (!n2)
    return t18 + "";
  var r2 = n2[0], i2 = n2[1], a2 = i2 - (Pi = 3 * Math.max(-8, Math.min(8, Math.floor(i2 / 3)))) + 1, o2 = r2.length;
  return a2 === o2 ? r2 : a2 > o2 ? r2 + new Array(a2 - o2 + 1).join("0") : a2 > 0 ? r2.slice(0, a2) + "." + r2.slice(a2) : "0." + new Array(1 - a2).join("0") + Ti(t18, Math.max(0, e2 + a2 - 1))[0];
}, X: (t18) => Math.round(t18).toString(16).toUpperCase(), x: (t18) => Math.round(t18).toString(16) };
function Ui(t18) {
  return t18;
}
var Fi, Bi, Di, Wi = Array.prototype.map, Hi = ["y", "z", "a", "f", "p", "n", "\xB5", "m", "", "k", "M", "G", "T", "P", "E", "Z", "Y"];
function Xi(t18) {
  var e2, n2, r2 = t18.grouping === void 0 || t18.thousands === void 0 ? Ui : (e2 = Wi.call(t18.grouping, Number), n2 = t18.thousands + "", (t19, r3) => {
    for (var i3 = t19.length, a3 = [], o3 = 0, u3 = e2[0], s3 = 0; i3 > 0 && u3 > 0 && (s3 + u3 + 1 > r3 && (u3 = Math.max(1, r3 - s3)), a3.push(t19.substring(i3 -= u3, i3 + u3)), !((s3 += u3 + 1) > r3)); )
      u3 = e2[o3 = (o3 + 1) % e2.length];
    return a3.reverse().join(n2);
  }), i2 = t18.currency === void 0 ? "" : t18.currency[0] + "", a2 = t18.currency === void 0 ? "" : t18.currency[1] + "", o2 = t18.decimal === void 0 ? "." : t18.decimal + "", u2 = t18.numerals === void 0 ? Ui : function(t19) {
    return (e3) => e3.replace(/[0-9]/g, (e4) => t19[+e4]);
  }(Wi.call(t18.numerals, String)), s2 = t18.percent === void 0 ? "%" : t18.percent + "", l2 = t18.minus === void 0 ? "-" : t18.minus + "", c2 = t18.nan === void 0 ? "NaN" : t18.nan + "";
  function h2(t19) {
    var e3 = (t19 = $i(t19)).fill, n3 = t19.align, h3 = t19.sign, f2 = t19.symbol, p2 = t19.zero, d2 = t19.width, v2 = t19.comma, g2 = t19.precision, y2 = t19.trim, m2 = t19.type;
    m2 === "n" ? (v2 = true, m2 = "g") : Vi[m2] || (g2 === void 0 && (g2 = 12), y2 = true, m2 = "g"), (p2 || e3 === "0" && n3 === "=") && (p2 = true, e3 = "0", n3 = "=");
    var _2 = f2 === "$" ? i2 : f2 === "#" && /[boxX]/.test(m2) ? "0" + m2.toLowerCase() : "", b2 = f2 === "$" ? a2 : /[%p]/.test(m2) ? s2 : "", w2 = Vi[m2], x2 = /[defgprs%]/.test(m2);
    function M2(t20) {
      var i3, a3, s3, f3 = _2, M3 = b2;
      if (m2 === "c")
        M3 = w2(t20) + M3, t20 = "";
      else {
        var S2 = (t20 = +t20) < 0 || 1 / t20 < 0;
        if (t20 = isNaN(t20) ? c2 : w2(Math.abs(t20), g2), y2 && (t20 = function(t21) {
          t:
            for (var e4, n4 = t21.length, r3 = 1, i4 = -1; r3 < n4; ++r3)
              switch (t21[r3]) {
                case ".":
                  i4 = e4 = r3;
                  break;
                case "0":
                  i4 === 0 && (i4 = r3), e4 = r3;
                  break;
                default:
                  if (!+t21[r3])
                    break t;
                  i4 > 0 && (i4 = 0);
              }
          return i4 > 0 ? t21.slice(0, i4) + t21.slice(e4 + 1) : t21;
        }(t20)), S2 && +t20 == 0 && h3 !== "+" && (S2 = false), f3 = (S2 ? h3 === "(" ? h3 : l2 : h3 === "-" || h3 === "(" ? "" : h3) + f3, M3 = (m2 === "s" ? Hi[8 + Pi / 3] : "") + M3 + (S2 && h3 === "(" ? ")" : ""), x2) {
          for (i3 = -1, a3 = t20.length; ++i3 < a3; )
            if (48 > (s3 = t20.charCodeAt(i3)) || s3 > 57) {
              M3 = (s3 === 46 ? o2 + t20.slice(i3 + 1) : t20.slice(i3)) + M3, t20 = t20.slice(0, i3);
              break;
            }
        }
      }
      v2 && !p2 && (t20 = r2(t20, 1 / 0));
      var A2 = f3.length + t20.length + M3.length, N2 = A2 < d2 ? new Array(d2 - A2 + 1).join(e3) : "";
      switch (v2 && p2 && (t20 = r2(N2 + t20, N2.length ? d2 - M3.length : 1 / 0), N2 = ""), n3) {
        case "<":
          t20 = f3 + t20 + M3 + N2;
          break;
        case "=":
          t20 = f3 + N2 + t20 + M3;
          break;
        case "^":
          t20 = N2.slice(0, A2 = N2.length >> 1) + f3 + t20 + M3 + N2.slice(A2);
          break;
        default:
          t20 = N2 + f3 + t20 + M3;
      }
      return u2(t20);
    }
    return g2 = g2 === void 0 ? 6 : /[gprs]/.test(m2) ? Math.max(1, Math.min(21, g2)) : Math.max(0, Math.min(20, g2)), M2.toString = () => t19 + "", M2;
  }
  return { format: h2, formatPrefix: function(t19, e3) {
    var n3 = h2(((t19 = $i(t19)).type = "f", t19)), r3 = 3 * Math.max(-8, Math.min(8, Math.floor(Li(e3) / 3))), i3 = Math.pow(10, -r3), a3 = Hi[8 + r3 / 3];
    return (t20) => n3(i3 * t20) + a3;
  } };
}
function Yi() {
  return Math.random();
}
Fi = Xi({ decimal: ".", thousands: ",", grouping: [3], currency: ["$", ""], minus: "-" }), Bi = Fi.format, Di = Fi.formatPrefix, function t12(e2) {
  function n2(t18, n3) {
    return t18 = t18 == null ? 0 : +t18, n3 = n3 == null ? 1 : +n3, arguments.length === 1 ? (n3 = t18, t18 = 0) : n3 -= t18, () => e2() * n3 + t18;
  }
  return n2.source = t12, n2;
}(Yi);
var Qi = function t13(e2) {
  function n2(t18, n3) {
    var r2, i2;
    return t18 = t18 == null ? 0 : +t18, n3 = n3 == null ? 1 : +n3, () => {
      var a2;
      if (r2 != null)
        a2 = r2, r2 = null;
      else
        do {
          r2 = 2 * e2() - 1, a2 = 2 * e2() - 1, i2 = r2 * r2 + a2 * a2;
        } while (!i2 || i2 > 1);
      return t18 + n3 * a2 * Math.sqrt(-2 * Math.log(i2) / i2);
    };
  }
  return n2.source = t13, n2;
}(Yi);
!function t14(e2) {
  function n2() {
    var t18 = Qi.source(e2).apply(this, arguments);
    return () => Math.exp(t18());
  }
  return n2.source = t14, n2;
}(Yi);
var Gi = function t15(e2) {
  function n2(t18) {
    return () => {
      for (var n3 = 0, r2 = 0; r2 < t18; ++r2)
        n3 += e2();
      return n3;
    };
  }
  return n2.source = t15, n2;
}(Yi);
function Zi(t18, e2) {
  switch (arguments.length) {
    case 0:
      break;
    case 1:
      this.range(t18);
      break;
    default:
      this.range(e2).domain(t18);
  }
  return this;
}
!function t16(e2) {
  function n2(t18) {
    var n3 = Gi.source(e2)(t18);
    return () => n3() / t18;
  }
  return n2.source = t16, n2;
}(Yi), function t17(e2) {
  function n2(t18) {
    return () => -Math.log(1 - e2()) / t18;
  }
  return n2.source = t17, n2;
}(Yi);
var Ji = Array.prototype, Ki = Ji.map, ta = Ji.slice;
function ea(t18) {
  return +t18;
}
var na = [0, 1];
function ra(t18) {
  return t18;
}
function ia(t18, e2) {
  return (e2 -= t18 = +t18) ? (n2) => (n2 - t18) / e2 : function(t19) {
    return () => t19;
  }(isNaN(e2) ? NaN : 0.5);
}
function aa(t18) {
  var e2, n2 = t18[0], r2 = t18[t18.length - 1];
  return n2 > r2 && (e2 = n2, n2 = r2, r2 = e2), (t19) => Math.max(n2, Math.min(r2, t19));
}
function oa(t18, e2, n2) {
  var r2 = t18[0], i2 = t18[1], a2 = e2[0], o2 = e2[1];
  return i2 < r2 ? (r2 = ia(i2, r2), a2 = n2(o2, a2)) : (r2 = ia(r2, i2), a2 = n2(a2, o2)), (t19) => a2(r2(t19));
}
function ua(t18, e2, n2) {
  var r2 = Math.min(t18.length, e2.length) - 1, i2 = new Array(r2), a2 = new Array(r2), o2 = -1;
  for (t18[r2] < t18[0] && (t18 = t18.slice().reverse(), e2 = e2.slice().reverse()); ++o2 < r2; )
    i2[o2] = ia(t18[o2], t18[o2 + 1]), a2[o2] = n2(e2[o2], e2[o2 + 1]);
  return (e3) => {
    var n3 = mt(t18, e3, 1, r2) - 1;
    return a2[n3](i2[n3](e3));
  };
}
function sa(t18, e2) {
  return e2.domain(t18.domain()).range(t18.range()).interpolate(t18.interpolate()).clamp(t18.clamp()).unknown(t18.unknown());
}
function la(t18, e2) {
  return function() {
    var t19, e3, n2, r2, i2, a2, o2 = na, u2 = na, s2 = kn, l2 = ra;
    function c2() {
      return r2 = Math.min(o2.length, u2.length) > 2 ? ua : oa, i2 = a2 = null, h2;
    }
    function h2(e4) {
      return isNaN(e4 = +e4) ? n2 : (i2 || (i2 = r2(o2.map(t19), u2, s2)))(t19(l2(e4)));
    }
    return h2.invert = (n3) => l2(e3((a2 || (a2 = r2(u2, o2.map(t19), Sn)))(n3))), h2.domain = function(t20) {
      return arguments.length ? (o2 = Ki.call(t20, ea), l2 === ra || (l2 = aa(o2)), c2()) : o2.slice();
    }, h2.range = function(t20) {
      return arguments.length ? (u2 = ta.call(t20), c2()) : u2.slice();
    }, h2.rangeRound = (t20) => (u2 = ta.call(t20), s2 = jn, c2()), h2.clamp = function(t20) {
      return arguments.length ? (l2 = t20 ? aa(o2) : ra, h2) : l2 !== ra;
    }, h2.interpolate = function(t20) {
      return arguments.length ? (s2 = t20, c2()) : s2;
    }, h2.unknown = function(t20) {
      return arguments.length ? (n2 = t20, h2) : n2;
    }, (n3, r3) => (t19 = n3, e3 = r3, c2());
  }()(t18, e2);
}
function ca(t18, e2, n2, r2) {
  var i2, a2 = function(t19, e3, n3) {
    var r3 = Math.abs(e3 - t19) / Math.max(0, n3), i3 = Math.pow(10, Math.floor(Math.log(r3) / Math.LN10)), a3 = r3 / i3;
    return a3 >= _t ? i3 *= 10 : a3 >= bt ? i3 *= 5 : a3 >= wt && (i3 *= 2), e3 < t19 ? -i3 : i3;
  }(t18, e2, n2);
  switch ((r2 = $i(r2 != null ? r2 : ",f")).type) {
    case "s":
      var o2 = Math.max(Math.abs(t18), Math.abs(e2));
      return r2.precision != null || isNaN(i2 = function(t19, e3) {
        return Math.max(0, 3 * Math.max(-8, Math.min(8, Math.floor(Li(e3) / 3))) - Li(Math.abs(t19)));
      }(a2, o2)) || (r2.precision = i2), Di(r2, o2);
    case "":
    case "e":
    case "g":
    case "p":
    case "r":
      r2.precision != null || isNaN(i2 = function(t19, e3) {
        return t19 = Math.abs(t19), e3 = Math.abs(e3) - t19, Math.max(0, Li(e3) - Li(t19)) + 1;
      }(a2, Math.max(Math.abs(t18), Math.abs(e2)))) || (r2.precision = i2 - (r2.type === "e"));
      break;
    case "f":
    case "%":
      r2.precision != null || isNaN(i2 = function(t19) {
        return Math.max(0, -Li(Math.abs(t19)));
      }(a2)) || (r2.precision = i2 - 2 * (r2.type === "%"));
  }
  return Bi(r2);
}
function ha(t18) {
  var e2 = t18.domain;
  return t18.ticks = (t19) => {
    var n2 = e2();
    return function(t20, e3, n3) {
      var r2, i2, a2, o2, u2 = -1;
      if (n3 = +n3, (t20 = +t20) == (e3 = +e3) && n3 > 0)
        return [t20];
      if ((r2 = e3 < t20) && (i2 = t20, t20 = e3, e3 = i2), (o2 = xt(t20, e3, n3)) === 0 || !isFinite(o2))
        return [];
      if (o2 > 0)
        for (t20 = Math.ceil(t20 / o2), e3 = Math.floor(e3 / o2), a2 = new Array(i2 = Math.ceil(e3 - t20 + 1)); ++u2 < i2; )
          a2[u2] = (t20 + u2) * o2;
      else
        for (t20 = Math.floor(t20 * o2), e3 = Math.ceil(e3 * o2), a2 = new Array(i2 = Math.ceil(t20 - e3 + 1)); ++u2 < i2; )
          a2[u2] = (t20 - u2) / o2;
      return r2 && a2.reverse(), a2;
    }(n2[0], n2[n2.length - 1], t19 != null ? t19 : 10);
  }, t18.tickFormat = (t19, n2) => {
    var r2 = e2();
    return ca(r2[0], r2[r2.length - 1], t19 != null ? t19 : 10, n2);
  }, t18.nice = (n2) => {
    n2 == null && (n2 = 10);
    var r2, i2 = e2(), a2 = 0, o2 = i2.length - 1, u2 = i2[a2], s2 = i2[o2];
    return s2 < u2 && (r2 = u2, u2 = s2, s2 = r2, r2 = a2, a2 = o2, o2 = r2), (r2 = xt(u2, s2, n2)) > 0 ? r2 = xt(u2 = Math.floor(u2 / r2) * r2, s2 = Math.ceil(s2 / r2) * r2, n2) : r2 < 0 && (r2 = xt(u2 = Math.ceil(u2 * r2) / r2, s2 = Math.floor(s2 * r2) / r2, n2)), r2 > 0 ? (i2[a2] = Math.floor(u2 / r2) * r2, i2[o2] = Math.ceil(s2 / r2) * r2, e2(i2)) : r2 < 0 && (i2[a2] = Math.ceil(u2 * r2) / r2, i2[o2] = Math.floor(s2 * r2) / r2, e2(i2)), t18;
  }, t18;
}
function fa() {
  var t18 = la(ra, ra);
  return t18.copy = () => sa(t18, fa()), Zi.apply(t18, arguments), ha(t18);
}
function pa() {
  var t18, e2 = 0, n2 = 1, r2 = 1, i2 = [0.5], a2 = [0, 1];
  function o2(e3) {
    return e3 <= e3 ? a2[mt(i2, e3, 0, r2)] : t18;
  }
  function u2() {
    var t19 = -1;
    for (i2 = new Array(r2); ++t19 < r2; )
      i2[t19] = ((t19 + 1) * n2 - (t19 - r2) * e2) / (r2 + 1);
    return o2;
  }
  return o2.domain = function(t19) {
    return arguments.length ? (e2 = +t19[0], n2 = +t19[1], u2()) : [e2, n2];
  }, o2.range = function(t19) {
    return arguments.length ? (r2 = (a2 = ta.call(t19)).length - 1, u2()) : a2.slice();
  }, o2.invertExtent = (t19) => {
    var o3 = a2.indexOf(t19);
    return o3 < 0 ? [NaN, NaN] : o3 < 1 ? [e2, i2[0]] : o3 >= r2 ? [i2[r2 - 1], n2] : [i2[o3 - 1], i2[o3]];
  }, o2.unknown = function(e3) {
    return arguments.length ? (t18 = e3, o2) : o2;
  }, o2.thresholds = () => i2.slice(), o2.copy = () => pa().domain([e2, n2]).range(a2).unknown(t18), Zi.apply(ha(o2), arguments);
}
function da(t18) {
  return function() {
    return t18;
  };
}
var va = Math.abs, ga = Math.atan2, ya = Math.cos, ma = Math.max, _a = Math.min, ba = Math.sin, wa = Math.sqrt, xa = Math.PI, Ma = xa / 2, Sa = 2 * xa;
function Aa(t18) {
  return t18 > 1 ? 0 : t18 < -1 ? xa : Math.acos(t18);
}
function Na(t18) {
  return t18 >= 1 ? Ma : t18 <= -1 ? -Ma : Math.asin(t18);
}
function Ca(t18) {
  return t18.innerRadius;
}
function Oa(t18) {
  return t18.outerRadius;
}
function ka(t18) {
  return t18.startAngle;
}
function ja(t18) {
  return t18.endAngle;
}
function Ea(t18) {
  return t18 && t18.padAngle;
}
function Ia(t18, e2, n2, r2, i2, a2, o2, u2) {
  var s2 = n2 - t18, l2 = r2 - e2, c2 = o2 - i2, h2 = u2 - a2, f2 = h2 * s2 - c2 * l2;
  if (!(f2 * f2 < 1e-12))
    return [t18 + (f2 = (c2 * (e2 - a2) - h2 * (t18 - i2)) / f2) * s2, e2 + f2 * l2];
}
function Ta(t18, e2, n2, r2, i2, a2, o2) {
  var u2 = t18 - n2, s2 = e2 - r2, l2 = (o2 ? a2 : -a2) / wa(u2 * u2 + s2 * s2), c2 = l2 * s2, h2 = -l2 * u2, f2 = t18 + c2, p2 = e2 + h2, d2 = n2 + c2, v2 = r2 + h2, g2 = (f2 + d2) / 2, y2 = (p2 + v2) / 2, m2 = d2 - f2, _2 = v2 - p2, b2 = m2 * m2 + _2 * _2, w2 = i2 - a2, x2 = f2 * v2 - d2 * p2, M2 = (_2 < 0 ? -1 : 1) * wa(ma(0, w2 * w2 * b2 - x2 * x2)), S2 = (x2 * _2 - m2 * M2) / b2, A2 = (-x2 * m2 - _2 * M2) / b2, N2 = (x2 * _2 + m2 * M2) / b2, C2 = (-x2 * m2 + _2 * M2) / b2, O2 = S2 - g2, k2 = A2 - y2, j2 = N2 - g2, E2 = C2 - y2;
  return O2 * O2 + k2 * k2 > j2 * j2 + E2 * E2 && (S2 = N2, A2 = C2), { cx: S2, cy: A2, x01: -c2, y01: -h2, x11: S2 * (i2 / w2 - 1), y11: A2 * (i2 / w2 - 1) };
}
function La(t18) {
  this._context = t18;
}
function Pa(t18) {
  return new La(t18);
}
function Ra(t18) {
  return t18[0];
}
function $a(t18) {
  return t18[1];
}
function za(t18) {
  return t18 < 0 ? -1 : 1;
}
function qa(t18, e2, n2) {
  var r2 = t18._x1 - t18._x0, i2 = e2 - t18._x1, a2 = (t18._y1 - t18._y0) / (r2 || i2 < 0 && -0), o2 = (n2 - t18._y1) / (i2 || r2 < 0 && -0), u2 = (a2 * i2 + o2 * r2) / (r2 + i2);
  return (za(a2) + za(o2)) * Math.min(Math.abs(a2), Math.abs(o2), 0.5 * Math.abs(u2)) || 0;
}
function Va(t18, e2) {
  var n2 = t18._x1 - t18._x0;
  return n2 ? (3 * (t18._y1 - t18._y0) / n2 - e2) / 2 : e2;
}
function Ua(t18, e2, n2) {
  var r2 = t18._x0, i2 = t18._y0, a2 = t18._x1, o2 = t18._y1, u2 = (a2 - r2) / 3;
  t18._context.bezierCurveTo(r2 + u2, i2 + u2 * e2, a2 - u2, o2 - u2 * n2, a2, o2);
}
function Fa(t18) {
  this._context = t18;
}
function Ba(t18) {
  return new Fa(t18);
}
function Da() {
  this._ = null;
}
function Wa(t18) {
  t18.U = t18.C = t18.L = t18.R = t18.P = t18.N = null;
}
function Ha(t18, e2) {
  var n2 = e2, r2 = e2.R, i2 = n2.U;
  i2 ? i2.L === n2 ? i2.L = r2 : i2.R = r2 : t18._ = r2, r2.U = i2, n2.U = r2, n2.R = r2.L, n2.R && (n2.R.U = n2), r2.L = n2;
}
function Xa(t18, e2) {
  var n2 = e2, r2 = e2.L, i2 = n2.U;
  i2 ? i2.L === n2 ? i2.L = r2 : i2.R = r2 : t18._ = r2, r2.U = i2, n2.U = r2, n2.L = r2.R, n2.L && (n2.L.U = n2), r2.R = n2;
}
function Ya(t18) {
  for (; t18.L; )
    t18 = t18.L;
  return t18;
}
function Qa(t18, e2, n2, r2) {
  var i2 = [null, null], a2 = bo.push(i2) - 1;
  return i2.left = t18, i2.right = e2, n2 && Za(i2, t18, e2, n2), r2 && Za(i2, e2, t18, r2), mo[t18.index].halfedges.push(a2), mo[e2.index].halfedges.push(a2), i2;
}
function Ga(t18, e2, n2) {
  var r2 = [e2, n2];
  return r2.left = t18, r2;
}
function Za(t18, e2, n2, r2) {
  t18[0] || t18[1] ? t18.left === n2 ? t18[1] = r2 : t18[0] = r2 : (t18[0] = r2, t18.left = e2, t18.right = n2);
}
function Ja(t18, e2, n2, r2, i2) {
  var a2, o2 = t18[0], u2 = t18[1], s2 = o2[0], l2 = o2[1], c2 = 0, h2 = 1, f2 = u2[0] - s2, p2 = u2[1] - l2;
  if (a2 = e2 - s2, f2 || !(a2 > 0)) {
    if (a2 /= f2, f2 < 0) {
      if (a2 < c2)
        return;
      a2 < h2 && (h2 = a2);
    } else if (f2 > 0) {
      if (a2 > h2)
        return;
      a2 > c2 && (c2 = a2);
    }
    if (a2 = r2 - s2, f2 || !(a2 < 0)) {
      if (a2 /= f2, f2 < 0) {
        if (a2 > h2)
          return;
        a2 > c2 && (c2 = a2);
      } else if (f2 > 0) {
        if (a2 < c2)
          return;
        a2 < h2 && (h2 = a2);
      }
      if (a2 = n2 - l2, p2 || !(a2 > 0)) {
        if (a2 /= p2, p2 < 0) {
          if (a2 < c2)
            return;
          a2 < h2 && (h2 = a2);
        } else if (p2 > 0) {
          if (a2 > h2)
            return;
          a2 > c2 && (c2 = a2);
        }
        if (a2 = i2 - l2, p2 || !(a2 < 0)) {
          if (a2 /= p2, p2 < 0) {
            if (a2 > h2)
              return;
            a2 > c2 && (c2 = a2);
          } else if (p2 > 0) {
            if (a2 < c2)
              return;
            a2 < h2 && (h2 = a2);
          }
          return !(c2 > 0 || h2 < 1) || (c2 > 0 && (t18[0] = [s2 + c2 * f2, l2 + c2 * p2]), h2 < 1 && (t18[1] = [s2 + h2 * f2, l2 + h2 * p2]), true);
        }
      }
    }
  }
}
function Ka(t18, e2, n2, r2, i2) {
  var a2 = t18[1];
  if (a2)
    return true;
  var o2, u2, s2 = t18[0], l2 = t18.left, c2 = t18.right, h2 = l2[0], f2 = l2[1], p2 = c2[0], d2 = c2[1], v2 = (h2 + p2) / 2, g2 = (f2 + d2) / 2;
  if (d2 === f2) {
    if (v2 < e2 || v2 >= r2)
      return;
    if (h2 > p2) {
      if (s2) {
        if (s2[1] >= i2)
          return;
      } else
        s2 = [v2, n2];
      a2 = [v2, i2];
    } else {
      if (s2) {
        if (s2[1] < n2)
          return;
      } else
        s2 = [v2, i2];
      a2 = [v2, n2];
    }
  } else if (u2 = g2 - (o2 = (h2 - p2) / (d2 - f2)) * v2, o2 < -1 || o2 > 1)
    if (h2 > p2) {
      if (s2) {
        if (s2[1] >= i2)
          return;
      } else
        s2 = [(n2 - u2) / o2, n2];
      a2 = [(i2 - u2) / o2, i2];
    } else {
      if (s2) {
        if (s2[1] < n2)
          return;
      } else
        s2 = [(i2 - u2) / o2, i2];
      a2 = [(n2 - u2) / o2, n2];
    }
  else if (f2 < d2) {
    if (s2) {
      if (s2[0] >= r2)
        return;
    } else
      s2 = [e2, o2 * e2 + u2];
    a2 = [r2, o2 * r2 + u2];
  } else {
    if (s2) {
      if (s2[0] < e2)
        return;
    } else
      s2 = [r2, o2 * r2 + u2];
    a2 = [e2, o2 * e2 + u2];
  }
  return t18[0] = s2, t18[1] = a2, true;
}
function to(t18, e2) {
  var n2 = t18.site, r2 = e2.left, i2 = e2.right;
  return n2 === i2 && (i2 = r2, r2 = n2), i2 ? Math.atan2(i2[1] - r2[1], i2[0] - r2[0]) : (n2 === r2 ? (r2 = e2[1], i2 = e2[0]) : (r2 = e2[0], i2 = e2[1]), Math.atan2(r2[0] - i2[0], i2[1] - r2[1]));
}
function eo(t18, e2) {
  return e2[+(e2.left !== t18.site)];
}
function no(t18, e2) {
  return e2[+(e2.left === t18.site)];
}
La.prototype = { areaStart: function() {
  this._line = 0;
}, areaEnd: function() {
  this._line = NaN;
}, lineStart: function() {
  this._point = 0;
}, lineEnd: function() {
  (this._line || this._line !== 0 && this._point === 1) && this._context.closePath(), this._line = 1 - this._line;
}, point: function(t18, e2) {
  switch (t18 = +t18, e2 = +e2, this._point) {
    case 0:
      this._point = 1, this._line ? this._context.lineTo(t18, e2) : this._context.moveTo(t18, e2);
      break;
    case 1:
      this._point = 2;
    default:
      this._context.lineTo(t18, e2);
  }
} }, Fa.prototype = { areaStart: function() {
  this._line = 0;
}, areaEnd: function() {
  this._line = NaN;
}, lineStart: function() {
  this._x0 = this._x1 = this._y0 = this._y1 = this._t0 = NaN, this._point = 0;
}, lineEnd: function() {
  switch (this._point) {
    case 2:
      this._context.lineTo(this._x1, this._y1);
      break;
    case 3:
      Ua(this, this._t0, Va(this, this._t0));
  }
  (this._line || this._line !== 0 && this._point === 1) && this._context.closePath(), this._line = 1 - this._line;
}, point: function(t18, e2) {
  var n2 = NaN;
  if (e2 = +e2, (t18 = +t18) !== this._x1 || e2 !== this._y1) {
    switch (this._point) {
      case 0:
        this._point = 1, this._line ? this._context.lineTo(t18, e2) : this._context.moveTo(t18, e2);
        break;
      case 1:
        this._point = 2;
        break;
      case 2:
        this._point = 3, Ua(this, Va(this, n2 = qa(this, t18, e2)), n2);
        break;
      default:
        Ua(this, this._t0, n2 = qa(this, t18, e2));
    }
    this._x0 = this._x1, this._x1 = t18, this._y0 = this._y1, this._y1 = e2, this._t0 = n2;
  }
} }, Object.create(Fa.prototype).point = function(t18, e2) {
  Fa.prototype.point.call(this, e2, t18);
}, Da.prototype = { constructor: Da, insert: function(t18, e2) {
  var n2, r2, i2;
  if (t18) {
    if (e2.P = t18, e2.N = t18.N, t18.N && (t18.N.P = e2), t18.N = e2, t18.R) {
      for (t18 = t18.R; t18.L; )
        t18 = t18.L;
      t18.L = e2;
    } else
      t18.R = e2;
    n2 = t18;
  } else
    this._ ? (t18 = Ya(this._), e2.P = null, e2.N = t18, t18.P = t18.L = e2, n2 = t18) : (e2.P = e2.N = null, this._ = e2, n2 = null);
  for (e2.L = e2.R = null, e2.U = n2, e2.C = true, t18 = e2; n2 && n2.C; )
    n2 === (r2 = n2.U).L ? (i2 = r2.R) && i2.C ? (n2.C = i2.C = false, r2.C = true, t18 = r2) : (t18 === n2.R && (Ha(this, n2), n2 = (t18 = n2).U), n2.C = false, r2.C = true, Xa(this, r2)) : (i2 = r2.L) && i2.C ? (n2.C = i2.C = false, r2.C = true, t18 = r2) : (t18 === n2.L && (Xa(this, n2), n2 = (t18 = n2).U), n2.C = false, r2.C = true, Ha(this, r2)), n2 = t18.U;
  this._.C = false;
}, remove: function(t18) {
  t18.N && (t18.N.P = t18.P), t18.P && (t18.P.N = t18.N), t18.N = t18.P = null;
  var e2, n2, r2, i2 = t18.U, a2 = t18.L, o2 = t18.R;
  if (n2 = a2 ? o2 ? Ya(o2) : a2 : o2, i2 ? i2.L === t18 ? i2.L = n2 : i2.R = n2 : this._ = n2, a2 && o2 ? (r2 = n2.C, n2.C = t18.C, n2.L = a2, a2.U = n2, n2 !== o2 ? (i2 = n2.U, n2.U = t18.U, t18 = n2.R, i2.L = t18, n2.R = o2, o2.U = n2) : (n2.U = i2, i2 = n2, t18 = n2.R)) : (r2 = t18.C, t18 = n2), t18 && (t18.U = i2), !r2)
    if (t18 && t18.C)
      t18.C = false;
    else {
      do {
        if (t18 === this._)
          break;
        if (t18 === i2.L) {
          if ((e2 = i2.R).C && (e2.C = false, i2.C = true, Ha(this, i2), e2 = i2.R), e2.L && e2.L.C || e2.R && e2.R.C) {
            e2.R && e2.R.C || (e2.L.C = false, e2.C = true, Xa(this, e2), e2 = i2.R), e2.C = i2.C, i2.C = e2.R.C = false, Ha(this, i2), t18 = this._;
            break;
          }
        } else if ((e2 = i2.L).C && (e2.C = false, i2.C = true, Xa(this, i2), e2 = i2.L), e2.L && e2.L.C || e2.R && e2.R.C) {
          e2.L && e2.L.C || (e2.R.C = false, e2.C = true, Ha(this, e2), e2 = i2.L), e2.C = i2.C, i2.C = e2.L.C = false, Xa(this, i2), t18 = this._;
          break;
        }
        e2.C = true, t18 = i2, i2 = i2.U;
      } while (!t18.C);
      t18 && (t18.C = false);
    }
} };
var ro, io = [];
function ao() {
  Wa(this), this.x = this.y = this.arc = this.site = this.cy = null;
}
function oo(t18) {
  var e2 = t18.P, n2 = t18.N;
  if (e2 && n2) {
    var r2 = e2.site, i2 = t18.site, a2 = n2.site;
    if (r2 !== a2) {
      var o2 = i2[0], u2 = i2[1], s2 = r2[0] - o2, l2 = r2[1] - u2, c2 = a2[0] - o2, h2 = a2[1] - u2, f2 = 2 * (s2 * h2 - l2 * c2);
      if (!(f2 >= -xo)) {
        var p2 = s2 * s2 + l2 * l2, d2 = c2 * c2 + h2 * h2, v2 = (h2 * p2 - l2 * d2) / f2, g2 = (s2 * d2 - c2 * p2) / f2, y2 = io.pop() || new ao();
        y2.arc = t18, y2.site = i2, y2.x = v2 + o2, y2.y = (y2.cy = g2 + u2) + Math.sqrt(v2 * v2 + g2 * g2), t18.circle = y2;
        for (var m2 = null, _2 = _o._; _2; )
          if (y2.y < _2.y || y2.y === _2.y && y2.x <= _2.x) {
            if (!_2.L) {
              m2 = _2.P;
              break;
            }
            _2 = _2.L;
          } else {
            if (!_2.R) {
              m2 = _2;
              break;
            }
            _2 = _2.R;
          }
        _o.insert(m2, y2), m2 || (ro = y2);
      }
    }
  }
}
function uo(t18) {
  var e2 = t18.circle;
  e2 && (e2.P || (ro = e2.N), _o.remove(e2), io.push(e2), Wa(e2), t18.circle = null);
}
var so = [];
function lo() {
  Wa(this), this.edge = this.site = this.circle = null;
}
function co(t18) {
  var e2 = so.pop() || new lo();
  return e2.site = t18, e2;
}
function ho(t18) {
  uo(t18), yo.remove(t18), so.push(t18), Wa(t18);
}
function fo(t18) {
  var e2 = t18.circle, n2 = e2.x, r2 = e2.cy, i2 = [n2, r2], a2 = t18.P, o2 = t18.N, u2 = [t18];
  ho(t18);
  for (var s2 = a2; s2.circle && Math.abs(n2 - s2.circle.x) < wo && Math.abs(r2 - s2.circle.cy) < wo; )
    a2 = s2.P, u2.unshift(s2), ho(s2), s2 = a2;
  u2.unshift(s2), uo(s2);
  for (var l2 = o2; l2.circle && Math.abs(n2 - l2.circle.x) < wo && Math.abs(r2 - l2.circle.cy) < wo; )
    o2 = l2.N, u2.push(l2), ho(l2), l2 = o2;
  u2.push(l2), uo(l2);
  var c2, h2 = u2.length;
  for (c2 = 1; c2 < h2; ++c2)
    l2 = u2[c2], s2 = u2[c2 - 1], Za(l2.edge, s2.site, l2.site, i2);
  s2 = u2[0], (l2 = u2[h2 - 1]).edge = Qa(s2.site, l2.site, null, i2), oo(s2), oo(l2);
}
function po(t18) {
  for (var e2, n2, r2, i2, a2 = t18[0], o2 = t18[1], u2 = yo._; u2; )
    if ((r2 = vo(u2, o2) - a2) > wo)
      u2 = u2.L;
    else {
      if (!((i2 = a2 - go(u2, o2)) > wo)) {
        r2 > -wo ? (e2 = u2.P, n2 = u2) : i2 > -wo ? (e2 = u2, n2 = u2.N) : e2 = n2 = u2;
        break;
      }
      if (!u2.R) {
        e2 = u2;
        break;
      }
      u2 = u2.R;
    }
  !function(t19) {
    mo[t19.index] = { site: t19, halfedges: [] };
  }(t18);
  var s2 = co(t18);
  if (yo.insert(e2, s2), e2 || n2) {
    if (e2 === n2)
      return uo(e2), n2 = co(e2.site), yo.insert(s2, n2), s2.edge = n2.edge = Qa(e2.site, s2.site), oo(e2), void oo(n2);
    if (n2) {
      uo(e2), uo(n2);
      var l2 = e2.site, c2 = l2[0], h2 = l2[1], f2 = t18[0] - c2, p2 = t18[1] - h2, d2 = n2.site, v2 = d2[0] - c2, g2 = d2[1] - h2, y2 = 2 * (f2 * g2 - p2 * v2), m2 = f2 * f2 + p2 * p2, _2 = v2 * v2 + g2 * g2, b2 = [(g2 * m2 - p2 * _2) / y2 + c2, (f2 * _2 - v2 * m2) / y2 + h2];
      Za(n2.edge, l2, d2, b2), s2.edge = Qa(l2, t18, null, b2), n2.edge = Qa(t18, d2, null, b2), oo(e2), oo(n2);
    } else
      s2.edge = Qa(e2.site, s2.site);
  }
}
function vo(t18, e2) {
  var n2 = t18.site, r2 = n2[0], i2 = n2[1], a2 = i2 - e2;
  if (!a2)
    return r2;
  var o2 = t18.P;
  if (!o2)
    return -1 / 0;
  var u2 = (n2 = o2.site)[0], s2 = n2[1], l2 = s2 - e2;
  if (!l2)
    return u2;
  var c2 = u2 - r2, h2 = 1 / a2 - 1 / l2, f2 = c2 / l2;
  return h2 ? (-f2 + Math.sqrt(f2 * f2 - 2 * h2 * (c2 * c2 / (-2 * l2) - s2 + l2 / 2 + i2 - a2 / 2))) / h2 + r2 : (r2 + u2) / 2;
}
function go(t18, e2) {
  var n2 = t18.N;
  if (n2)
    return vo(n2, e2);
  var r2 = t18.site;
  return r2[1] === e2 ? r2[0] : 1 / 0;
}
var yo, mo, _o, bo, wo = 1e-6, xo = 1e-12;
function Mo(t18, e2) {
  return e2[1] - t18[1] || e2[0] - t18[0];
}
function So(t18, e2) {
  var n2, r2, i2, a2 = t18.sort(Mo).pop();
  for (bo = [], mo = new Array(t18.length), yo = new Da(), _o = new Da(); ; )
    if (i2 = ro, a2 && (!i2 || a2[1] < i2.y || a2[1] === i2.y && a2[0] < i2.x))
      a2[0] === n2 && a2[1] === r2 || (po(a2), n2 = a2[0], r2 = a2[1]), a2 = t18.pop();
    else {
      if (!i2)
        break;
      fo(i2.arc);
    }
  if (function() {
    for (var t19, e3, n3, r3, i3 = 0, a3 = mo.length; i3 < a3; ++i3)
      if ((t19 = mo[i3]) && (r3 = (e3 = t19.halfedges).length)) {
        var o3 = new Array(r3), u3 = new Array(r3);
        for (n3 = 0; n3 < r3; ++n3)
          o3[n3] = n3, u3[n3] = to(t19, bo[e3[n3]]);
        for (o3.sort((t20, e4) => u3[e4] - u3[t20]), n3 = 0; n3 < r3; ++n3)
          u3[n3] = e3[o3[n3]];
        for (n3 = 0; n3 < r3; ++n3)
          e3[n3] = u3[n3];
      }
  }(), e2) {
    var o2 = +e2[0][0], u2 = +e2[0][1], s2 = +e2[1][0], l2 = +e2[1][1];
    !function(t19, e3, n3, r3) {
      for (var i3, a3 = bo.length; a3--; )
        Ka(i3 = bo[a3], t19, e3, n3, r3) && Ja(i3, t19, e3, n3, r3) && (Math.abs(i3[0][0] - i3[1][0]) > wo || Math.abs(i3[0][1] - i3[1][1]) > wo) || delete bo[a3];
    }(o2, u2, s2, l2), function(t19, e3, n3, r3) {
      var i3, a3, o3, u3, s3, l3, c2, h2, f2, p2, d2, v2, g2 = mo.length, y2 = true;
      for (i3 = 0; i3 < g2; ++i3)
        if (a3 = mo[i3]) {
          for (o3 = a3.site, u3 = (s3 = a3.halfedges).length; u3--; )
            bo[s3[u3]] || s3.splice(u3, 1);
          for (u3 = 0, l3 = s3.length; u3 < l3; )
            d2 = (p2 = no(a3, bo[s3[u3]]))[0], v2 = p2[1], h2 = (c2 = eo(a3, bo[s3[++u3 % l3]]))[0], f2 = c2[1], (Math.abs(d2 - h2) > wo || Math.abs(v2 - f2) > wo) && (s3.splice(u3, 0, bo.push(Ga(o3, p2, Math.abs(d2 - t19) < wo && r3 - v2 > wo ? [t19, Math.abs(h2 - t19) < wo ? f2 : r3] : Math.abs(v2 - r3) < wo && n3 - d2 > wo ? [Math.abs(f2 - r3) < wo ? h2 : n3, r3] : Math.abs(d2 - n3) < wo && v2 - e3 > wo ? [n3, Math.abs(h2 - n3) < wo ? f2 : e3] : Math.abs(v2 - e3) < wo && d2 - t19 > wo ? [Math.abs(f2 - e3) < wo ? h2 : t19, e3] : null)) - 1), ++l3);
          l3 && (y2 = false);
        }
      if (y2) {
        var m2, _2, b2, w2 = 1 / 0;
        for (i3 = 0, y2 = null; i3 < g2; ++i3)
          (a3 = mo[i3]) && (b2 = (m2 = (o3 = a3.site)[0] - t19) * m2 + (_2 = o3[1] - e3) * _2) < w2 && (w2 = b2, y2 = a3);
        if (y2) {
          var x2 = [t19, e3], M2 = [t19, r3], S2 = [n3, r3], A2 = [n3, e3];
          y2.halfedges.push(bo.push(Ga(o3 = y2.site, x2, M2)) - 1, bo.push(Ga(o3, M2, S2)) - 1, bo.push(Ga(o3, S2, A2)) - 1, bo.push(Ga(o3, A2, x2)) - 1);
        }
      }
      for (i3 = 0; i3 < g2; ++i3)
        (a3 = mo[i3]) && (a3.halfedges.length || delete mo[i3]);
    }(o2, u2, s2, l2);
  }
  this.edges = bo, this.cells = mo, yo = _o = bo = mo = null;
}
function Ao(t18, e2, n2) {
  this.k = t18, this.x = e2, this.y = n2;
}
So.prototype = { constructor: So, polygons: function() {
  var t18 = this.edges;
  return this.cells.map((e2) => {
    var n2 = e2.halfedges.map((n3) => eo(e2, t18[n3]));
    return n2.data = e2.site.data, n2;
  });
}, triangles: function() {
  var t18 = [], e2 = this.edges;
  return this.cells.forEach((n2, r2) => {
    if (a2 = (i2 = n2.halfedges).length)
      for (var i2, a2, o2, u2, s2, l2, c2 = n2.site, h2 = -1, f2 = e2[i2[a2 - 1]], p2 = f2.left === c2 ? f2.right : f2.left; ++h2 < a2; )
        o2 = p2, p2 = (f2 = e2[i2[h2]]).left === c2 ? f2.right : f2.left, o2 && p2 && r2 < o2.index && r2 < p2.index && (s2 = o2, l2 = p2, ((u2 = c2)[0] - l2[0]) * (s2[1] - u2[1]) - (u2[0] - s2[0]) * (l2[1] - u2[1]) < 0) && t18.push([c2.data, o2.data, p2.data]);
  }), t18;
}, links: function() {
  return this.edges.filter((t18) => t18.right).map((t18) => ({ source: t18.left.data, target: t18.right.data }));
}, find: function(t18, e2, n2) {
  for (var r2, i2, a2 = this, o2 = a2._found || 0, u2 = a2.cells.length; !(i2 = a2.cells[o2]); )
    if (++o2 >= u2)
      return null;
  var s2 = t18 - i2.site[0], l2 = e2 - i2.site[1], c2 = s2 * s2 + l2 * l2;
  do {
    i2 = a2.cells[r2 = o2], o2 = null, i2.halfedges.forEach((n3) => {
      var r3 = a2.edges[n3], u3 = r3.left;
      if (u3 !== i2.site && u3 || (u3 = r3.right)) {
        var s3 = t18 - u3[0], l3 = e2 - u3[1], h2 = s3 * s3 + l3 * l3;
        h2 < c2 && (c2 = h2, o2 = u3.index);
      }
    });
  } while (o2 !== null);
  return a2._found = r2, n2 == null || c2 <= n2 * n2 ? i2.site : null;
} }, Ao.prototype = { constructor: Ao, scale: function(t18) {
  return t18 === 1 ? this : new Ao(this.k * t18, this.x, this.y);
}, translate: function(t18, e2) {
  return t18 === 0 & e2 === 0 ? this : new Ao(this.k, this.x + this.k * t18, this.y + this.k * e2);
}, apply: function(t18) {
  return [t18[0] * this.k + this.x, t18[1] * this.k + this.y];
}, applyX: function(t18) {
  return t18 * this.k + this.x;
}, applyY: function(t18) {
  return t18 * this.k + this.y;
}, invert: function(t18) {
  return [(t18[0] - this.x) / this.k, (t18[1] - this.y) / this.k];
}, invertX: function(t18) {
  return (t18 - this.x) / this.k;
}, invertY: function(t18) {
  return (t18 - this.y) / this.k;
}, rescaleX: function(t18) {
  return t18.copy().domain(t18.range().map(this.invertX, this).map(t18.invert, t18));
}, rescaleY: function(t18) {
  return t18.copy().domain(t18.range().map(this.invertY, this).map(t18.invert, t18));
}, toString: function() {
  return "translate(" + this.x + "," + this.y + ") scale(" + this.k + ")";
} }, new Ao(1, 0, 0), Ao.prototype;
var No, Co = r["__core-js_shared__"], Oo = (No = /[^.]+$/.exec(Co && Co.keys && Co.keys.IE_PROTO || "")) ? "Symbol(src)_1." + No : "";
var ko = Function.prototype.toString;
function jo(t18) {
  if (t18 != null) {
    try {
      return ko.call(t18);
    } catch (t19) {
    }
    try {
      return t18 + "";
    } catch (t19) {
    }
  }
  return "";
}
var Eo = /^\[object .+?Constructor\]$/, Io = Function.prototype, To = Object.prototype, Lo = Io.toString, Po = To.hasOwnProperty, Ro = RegExp("^" + Lo.call(Po).replace(/[\\^$.*+?()[\]{}|]/g, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$");
function $o(e2) {
  return !(!t(e2) || function(t18) {
    return !!Oo && Oo in t18;
  }(e2)) && (at(e2) ? Ro : Eo).test(jo(e2));
}
function zo(t18, e2) {
  var n2 = function(t19, e3) {
    return t19 == null ? void 0 : t19[e3];
  }(t18, e2);
  return $o(n2) ? n2 : void 0;
}
var qo = zo(r, "DataView"), Vo = zo(r, "Map"), Uo = zo(r, "Promise"), Fo = zo(r, "Set"), Bo = zo(r, "WeakMap"), Do = jo(qo), Wo = jo(Vo), Ho = jo(Uo), Xo = jo(Fo), Yo = jo(Bo), Qo = v;
(qo && Qo(new qo(new ArrayBuffer(1))) != "[object DataView]" || Vo && Qo(new Vo()) != "[object Map]" || Uo && Qo(Uo.resolve()) != "[object Promise]" || Fo && Qo(new Fo()) != "[object Set]" || Bo && Qo(new Bo()) != "[object WeakMap]") && (Qo = (t18) => {
  var e2 = v(t18), n2 = e2 == "[object Object]" ? t18.constructor : void 0, r2 = n2 ? jo(n2) : "";
  if (r2)
    switch (r2) {
      case Do:
        return "[object DataView]";
      case Wo:
        return "[object Map]";
      case Ho:
        return "[object Promise]";
      case Xo:
        return "[object Set]";
      case Yo:
        return "[object WeakMap]";
    }
  return e2;
});
var Go = Qo, Zo = Object.prototype.hasOwnProperty;
function Jo(t18) {
  if (t18 == null)
    return true;
  if (ot(t18) && (R(t18) || typeof t18 == "string" || typeof t18.splice == "function" || V(t18) || Z(t18) || P(t18)))
    return !t18.length;
  var e2 = Go(t18);
  if (e2 == "[object Map]" || e2 == "[object Set]")
    return !t18.size;
  if (tt(t18))
    return !it(t18).length;
  for (var n2 in t18)
    if (Zo.call(t18, n2))
      return false;
  return true;
}
function Ko(t18) {
  var e2 = t18 == null ? 0 : t18.length;
  return e2 ? t18[e2 - 1] : void 0;
}
function tu(t18, e2, n2) {
  var r2 = -1, i2 = t18.length;
  e2 < 0 && (e2 = -e2 > i2 ? 0 : i2 + e2), (n2 = n2 > i2 ? i2 : n2) < 0 && (n2 += i2), i2 = e2 > n2 ? 0 : n2 - e2 >>> 0, e2 >>>= 0;
  for (var a2 = Array(i2); ++r2 < i2; )
    a2[r2] = t18[r2 + e2];
  return a2;
}
function eu(t18) {
  var e2 = function(t19) {
    return t19 ? 1 / 0 === (t19 = x(t19)) || -1 / 0 === t19 ? 17976931348623157e292 * (t19 < 0 ? -1 : 1) : t19 == t19 ? t19 : 0 : t19 === 0 ? t19 : 0;
  }(t18), n2 = e2 % 1;
  return e2 == e2 ? n2 ? e2 - n2 : e2 : 0;
}
var nu = Math.min;
function ru(t18, e2) {
  return function(t19) {
    return t19 && t19.length ? ((t20, e3) => {
      for (var n2, r2 = -1, i2 = t20.length; ++r2 < i2; ) {
        var a2 = e3(t20[r2]);
        a2 !== void 0 && (n2 = n2 === void 0 ? a2 : n2 + a2);
      }
      return n2;
    })(t19, ht) : 0;
  }(function(t19, e3, n2) {
    return t19 && t19.length ? tu(t19, 0, (e3 = n2 || e3 === void 0 ? 1 : eu(e3)) < 0 ? 0 : e3) : [];
  }(t18, e2));
}
function iu({ tickData: t18, customSegmentStops: e2, min: n2, max: r2 }) {
  if (!R(e2) || Jo(e2))
    return t18;
  if (function(t19) {
    return t19 && t19.length ? t19[0] : void 0;
  }(e2) !== n2)
    throw new Error(`First value should be equivalent to min value given. Current min value - ${n2}`);
  if (Ko(e2) !== r2)
    throw new Error(`Last value should be equivalent to max value given. Current min value - ${r2}`);
  const i2 = e2.map((t19, i3) => {
    if (i3 !== 0)
      return (t19 - e2[i3 - 1]) / (r2 - n2);
  });
  return function(t19, e3, n3) {
    var r3 = t19 == null ? 0 : t19.length;
    return r3 ? tu(t19, (e3 = n3 || e3 === void 0 ? 1 : eu(e3)) < 0 ? 0 : e3, r3) : [];
  }(i2);
}
function au({ maxSegmentLabelCount: t18, segmentCount: e2 }) {
  const n2 = parseInt(t18, 10), r2 = parseInt(e2, 10);
  return (typeof (i2 = n2) == "number" || g(i2) && v(i2) == "[object Number]") && n2 >= 0 && n2 <= r2 ? n2 : r2;
  var i2;
}
function ou({ min: t18, max: e2, segments: n2 }) {
  return fa().range([0, 1]).domain([t18, e2]);
}
function uu(t18, { min: e2, max: n2, segments: r2 }) {
  let i2 = [];
  i2 = t18.ticks(r2);
  if ((Ko(i2) !== n2 || r2 < i2.length) && i2.length > 1) {
    const t19 = (n2 - e2) / r2;
    i2 = [e2], function(t20, e3) {
      if ((t20 = eu(t20)) < 1 || t20 > 9007199254740991)
        return [];
      var n3 = 4294967295, r3 = nu(t20, 4294967295);
      t20 -= 4294967295;
      for (var i3 = j(r3, e3 = ft(e3)); ++n3 < t20; )
        e3(n3);
    }(r2, (n3) => {
      i2.push(e2 + (n3 + 1) * t19);
    });
  }
  return i2.length === 1 && (i2 = [e2, n2]), i2;
}
function su(t18) {
  return t18 * Math.PI / 180;
}
const lu = { Inside: "INSIDE", Outside: "OUTSIDE" }, cu = { easeLinear: "easeLinear", easeQuadIn: "easeQuadIn", easeQuadOut: "easeQuadOut", easeQuadInOut: "easeQuadInOut", easeCubicIn: "easeCubicIn", easeCubicOut: "easeCubicOut", easeCubicInOut: "easeCubicInOut", easePolyIn: "easePolyIn", easePolyOut: "easePolyOut", easePolyInOut: "easePolyInOut", easeSinIn: "easeSinIn", easeSinOut: "easeSinOut", easeSinInOut: "easeSinInOut", easeExpIn: "easeExpIn", easeExpOut: "easeExpOut", easeExpInOut: "easeExpInOut", easeCircleIn: "easeCircleIn", easeCircleOut: "easeCircleOut", easeCircleInOut: "easeCircleInOut", easeBounceIn: "easeBounceIn", easeBounceOut: "easeBounceOut", easeBounceInOut: "easeBounceInOut", easeBackIn: "easeBackIn", easeBackOut: "easeBackOut", easeBackInOut: "easeBackInOut", easeElasticIn: "easeElasticIn", easeElasticOut: "easeElasticOut", easeElasticInOut: "easeElasticInOut", easeElastic: "easeElastic" }, hu = { value: 0, minValue: 0, maxValue: 1e3, forceRender: false, width: 300, height: 300, paddingHorizontal: 0, paddingVertical: 0, fluidWidth: false, dimensionUnit: "px", segments: 5, maxSegmentLabels: -1, customSegmentStops: [], customSegmentLabels: [], needleColor: "steelblue", startColor: "#FF471A", endColor: "#33CC33", segmentColors: [], needleTransition: cu.easeQuadInOut, needleTransitionDuration: 500, needleHeightRatio: 0.9, ringWidth: 60, textColor: "#666", valueFormat: "", currentValueText: "${value}", currentValuePlaceholderStyle: "${value}", labelFontSize: "14px", valueTextFontSize: "16px", valueTextFontWeight: "bold", svgAriaLabel: "React d3 speedometer" }, fu = { ringInset: 20, pointerWidth: 10, pointerTailLength: 5, minAngle: -90, maxAngle: 90, labelInset: 10 };
var pu = Number.isNaN || function(t18) {
  return typeof t18 == "number" && t18 != t18;
};
function du(t18, e2) {
  if (t18.length !== e2.length)
    return false;
  for (var n2 = 0; n2 < t18.length; n2++)
    if (r2 = t18[n2], i2 = e2[n2], !(r2 === i2 || pu(r2) && pu(i2)))
      return false;
  var r2, i2;
  return true;
}
function vu(t18, e2) {
  var n2;
  e2 === void 0 && (e2 = du);
  var r2, i2 = [], a2 = false;
  return function() {
    for (var o2 = [], u2 = 0; u2 < arguments.length; u2++)
      o2[u2] = arguments[u2];
    return a2 && n2 === this && e2(o2, i2) || (r2 = t18.apply(this, o2), a2 = true, n2 = this, i2 = o2), r2;
  };
}
const gu = vu(function(t18) {
  return ou({ min: t18.minValue, max: t18.maxValue, segments: t18.maxSegmentLabels });
}), yu = vu(function(t18) {
  let e2 = uu(gu(t18), { min: t18.minValue, max: t18.maxValue, segments: t18.maxSegmentLabels });
  t18.customSegmentStops.length > 0 && t18.maxSegmentLabels !== 0 && (e2 = t18.customSegmentStops);
  return e2;
}), mu = vu(function(t18) {
  const e2 = function(t19, e3, n2) {
    t19 = +t19, e3 = +e3, n2 = (i2 = arguments.length) < 2 ? (e3 = t19, t19 = 0, 1) : i2 < 3 ? 1 : +n2;
    for (var r2 = -1, i2 = 0 | Math.max(0, Math.ceil((e3 - t19) / n2)), a2 = new Array(i2); ++r2 < i2; )
      a2[r2] = t19 + r2 * n2;
    return a2;
  }(t18.majorTicks).map((e3) => 1 / t18.majorTicks);
  return iu({ tickData: e2, customSegmentStops: t18.customSegmentStops, min: t18.minValue, max: t18.maxValue });
}), _u = vu(function(t18) {
  const e2 = mu(t18), n2 = t18.maxAngle - t18.minAngle, r2 = t18.width / 2;
  return function() {
    var t19 = Ca, e3 = Oa, n3 = da(0), r3 = null, i2 = ka, a2 = ja, o2 = Ea, u2 = null;
    function s2() {
      var s3, l2, c2 = +t19.apply(this, arguments), h2 = +e3.apply(this, arguments), f2 = i2.apply(this, arguments) - Ma, p2 = a2.apply(this, arguments) - Ma, d2 = va(p2 - f2), v2 = p2 > f2;
      if (u2 || (u2 = s3 = Oi()), h2 < c2 && (l2 = h2, h2 = c2, c2 = l2), h2 > 1e-12)
        if (d2 > Sa - 1e-12)
          u2.moveTo(h2 * ya(f2), h2 * ba(f2)), u2.arc(0, 0, h2, f2, p2, !v2), c2 > 1e-12 && (u2.moveTo(c2 * ya(p2), c2 * ba(p2)), u2.arc(0, 0, c2, p2, f2, v2));
        else {
          var g2, y2, m2 = f2, _2 = p2, b2 = f2, w2 = p2, x2 = d2, M2 = d2, S2 = o2.apply(this, arguments) / 2, A2 = S2 > 1e-12 && (r3 ? +r3.apply(this, arguments) : wa(c2 * c2 + h2 * h2)), N2 = _a(va(h2 - c2) / 2, +n3.apply(this, arguments)), C2 = N2, O2 = N2;
          if (A2 > 1e-12) {
            var k2 = Na(A2 / c2 * ba(S2)), j2 = Na(A2 / h2 * ba(S2));
            (x2 -= 2 * k2) > 1e-12 ? (b2 += k2 *= v2 ? 1 : -1, w2 -= k2) : (x2 = 0, b2 = w2 = (f2 + p2) / 2), (M2 -= 2 * j2) > 1e-12 ? (m2 += j2 *= v2 ? 1 : -1, _2 -= j2) : (M2 = 0, m2 = _2 = (f2 + p2) / 2);
          }
          var E2 = h2 * ya(m2), I2 = h2 * ba(m2), T2 = c2 * ya(w2), L2 = c2 * ba(w2);
          if (N2 > 1e-12) {
            var P2, R2 = h2 * ya(_2), $2 = h2 * ba(_2), z2 = c2 * ya(b2), q2 = c2 * ba(b2);
            if (d2 < xa && (P2 = Ia(E2, I2, z2, q2, R2, $2, T2, L2))) {
              var V2 = E2 - P2[0], U2 = I2 - P2[1], F2 = R2 - P2[0], B2 = $2 - P2[1], D2 = 1 / ba(Aa((V2 * F2 + U2 * B2) / (wa(V2 * V2 + U2 * U2) * wa(F2 * F2 + B2 * B2))) / 2), W2 = wa(P2[0] * P2[0] + P2[1] * P2[1]);
              C2 = _a(N2, (c2 - W2) / (D2 - 1)), O2 = _a(N2, (h2 - W2) / (D2 + 1));
            }
          }
          M2 > 1e-12 ? O2 > 1e-12 ? (g2 = Ta(z2, q2, E2, I2, h2, O2, v2), y2 = Ta(R2, $2, T2, L2, h2, O2, v2), u2.moveTo(g2.cx + g2.x01, g2.cy + g2.y01), O2 < N2 ? u2.arc(g2.cx, g2.cy, O2, ga(g2.y01, g2.x01), ga(y2.y01, y2.x01), !v2) : (u2.arc(g2.cx, g2.cy, O2, ga(g2.y01, g2.x01), ga(g2.y11, g2.x11), !v2), u2.arc(0, 0, h2, ga(g2.cy + g2.y11, g2.cx + g2.x11), ga(y2.cy + y2.y11, y2.cx + y2.x11), !v2), u2.arc(y2.cx, y2.cy, O2, ga(y2.y11, y2.x11), ga(y2.y01, y2.x01), !v2))) : (u2.moveTo(E2, I2), u2.arc(0, 0, h2, m2, _2, !v2)) : u2.moveTo(E2, I2), c2 > 1e-12 && x2 > 1e-12 ? C2 > 1e-12 ? (g2 = Ta(T2, L2, R2, $2, c2, -C2, v2), y2 = Ta(E2, I2, z2, q2, c2, -C2, v2), u2.lineTo(g2.cx + g2.x01, g2.cy + g2.y01), C2 < N2 ? u2.arc(g2.cx, g2.cy, C2, ga(g2.y01, g2.x01), ga(y2.y01, y2.x01), !v2) : (u2.arc(g2.cx, g2.cy, C2, ga(g2.y01, g2.x01), ga(g2.y11, g2.x11), !v2), u2.arc(0, 0, c2, ga(g2.cy + g2.y11, g2.cx + g2.x11), ga(y2.cy + y2.y11, y2.cx + y2.x11), v2), u2.arc(y2.cx, y2.cy, C2, ga(y2.y11, y2.x11), ga(y2.y01, y2.x01), !v2))) : u2.arc(0, 0, c2, w2, b2, v2) : u2.lineTo(T2, L2);
        }
      else
        u2.moveTo(0, 0);
      if (u2.closePath(), s3)
        return u2 = null, s3 + "" || null;
    }
    return s2.centroid = function() {
      var n4 = (+t19.apply(this, arguments) + +e3.apply(this, arguments)) / 2, r4 = (+i2.apply(this, arguments) + +a2.apply(this, arguments)) / 2 - xa / 2;
      return [ya(r4) * n4, ba(r4) * n4];
    }, s2.innerRadius = function(e4) {
      return arguments.length ? (t19 = typeof e4 == "function" ? e4 : da(+e4), s2) : t19;
    }, s2.outerRadius = function(t20) {
      return arguments.length ? (e3 = typeof t20 == "function" ? t20 : da(+t20), s2) : e3;
    }, s2.cornerRadius = function(t20) {
      return arguments.length ? (n3 = typeof t20 == "function" ? t20 : da(+t20), s2) : n3;
    }, s2.padRadius = function(t20) {
      return arguments.length ? (r3 = t20 == null ? null : typeof t20 == "function" ? t20 : da(+t20), s2) : r3;
    }, s2.startAngle = function(t20) {
      return arguments.length ? (i2 = typeof t20 == "function" ? t20 : da(+t20), s2) : i2;
    }, s2.endAngle = function(t20) {
      return arguments.length ? (a2 = typeof t20 == "function" ? t20 : da(+t20), s2) : a2;
    }, s2.padAngle = function(t20) {
      return arguments.length ? (o2 = typeof t20 == "function" ? t20 : da(+t20), s2) : o2;
    }, s2.context = function(t20) {
      return arguments.length ? (u2 = t20 != null ? t20 : null, s2) : u2;
    }, s2;
  }().innerRadius(r2 - t18.ringWidth - t18.ringInset).outerRadius(r2 - t18.ringInset).startAngle((r3, i2) => {
    const a2 = ru(e2, i2);
    return su(t18.minAngle + a2 * n2);
  }).endAngle((r3, i2) => {
    const a2 = ru(e2, i2 + 1);
    return su(t18.minAngle + a2 * n2);
  });
});
const bu = ({ d3_refs: t18, newValue: e2, config: n2 }) => {
  const r2 = gu(n2)(e2), i2 = n2.maxAngle - n2.minAngle, a2 = n2.minAngle + r2 * i2;
  t18.pointer.transition().duration(n2.needleTransitionDuration).ease(function(t19) {
    switch (t19) {
      case cu.easeLinear:
        return Fr;
      case cu.easeQuadIn:
        return Br;
      case cu.easeQuadOut:
        return Dr;
      case cu.easeQuadInOut:
        return Wr;
      case cu.easeCubicIn:
        return Hr;
      case cu.easeCubicOut:
        return Xr;
      case cu.easeCubicInOut:
        return Yr;
      case cu.easePolyIn:
        return Qr;
      case cu.easePolyOut:
        return Gr;
      case cu.easePolyInOut:
        return Zr;
      case cu.easeSinIn:
        return ti;
      case cu.easeSinOut:
        return ei;
      case cu.easeSinInOut:
        return ni;
      case cu.easeExpIn:
        return ii;
      case cu.easeExpOut:
        return ai;
      case cu.easeExpInOut:
        return oi;
      case cu.easeCircleIn:
        return ui;
      case cu.easeCircleOut:
        return si;
      case cu.easeCircleInOut:
        return li;
      case cu.easeBounceIn:
        return fi;
      case cu.easeBounceOut:
        return pi;
      case cu.easeBounceInOut:
        return di;
      case cu.easeBackIn:
        return vi;
      case cu.easeBackOut:
        return gi;
      case cu.easeBackInOut:
        return yi;
      case cu.easeElasticIn:
        return _i;
      case cu.easeElasticOut:
        return bi;
      case cu.easeElasticInOut:
        return wi;
      case cu.easeElastic:
        return bi;
      default:
        return Wr;
    }
  }(n2.needleTransition)).attr("transform", `rotate(${a2})`), t18.current_value_text.text(function(t19, e3) {
    const n3 = e3.labelFormat(t19), r3 = e3.currentValuePlaceholderStyle;
    return e3.currentValueText.replace(r3, n3);
  }(e2, n2));
}, wu = ({ container: t18, config: e2 }) => {
  const n2 = function(t19) {
    return t19.width / 2;
  }(e2), r2 = function(t19, e3, n3) {
    return `translate(${t19 + e3}, ${t19 + n3})`;
  }(n2, e2.paddingHorizontal, e2.paddingVertical), i2 = function({ container: t19, config: e3 }) {
    const n3 = e3.width + 2 * e3.paddingHorizontal, r3 = e3.height + 2 * e3.paddingVertical;
    return $e(t19).append("svg:svg").attr("class", "speedometer").attr("width", `${n3}${e3.dimensionUnit}`).attr("height", `${r3}${e3.dimensionUnit}`).attr("role", "img").attr("focusable", "false").attr("aria-label", e3.svgAriaLabel).style("width", `${n3}${e3.dimensionUnit}`).style("height", `${r3}${e3.dimensionUnit}`);
  }({ container: t18, config: e2 });
  return function({ config: t19, svg: e3, centerTx: n3 }) {
    const r3 = mu(t19), i3 = _u(t19);
    e3.append("g").attr("class", "arc").attr("transform", n3).selectAll("path").data(r3).enter().append("path").attr("class", "speedo-segment").attr("fill", (e4, n4) => !Jo(t19.segmentColors) && t19.segmentColors[n4] ? t19.segmentColors[n4] : t19.arcColorFn(e4 * n4)).attr("d", i3);
  }({ config: e2, svg: i2, centerTx: r2 }), function({ config: t19, svg: e3, centerTx: n3, r: r3 }) {
    const i3 = yu(t19), a2 = mu(t19), o2 = gu(t19), u2 = t19.maxAngle - t19.minAngle, { customSegmentLabels: s2 } = t19, l2 = R(s2) && !Jo(s2), c2 = l2 && s2.length === a2.length;
    if (l2 && !c2)
      throw new Error(`Custom Segment Labels should be an array with length of ${a2.length}`);
    if (l2 && c2)
      return void function({ config: t20, svg: e4, centerTx: n4, r: r4, ticks: i4, tickData: a3, scale: o3, range: u3 }) {
        const { customSegmentStops: s3, customSegmentLabels: l3 } = t20;
        function c3(e5, n5) {
          const r5 = s3.length === 0 ? o3(e5) : ru(a3, n5);
          return t20.minAngle + r5 * u3;
        }
        const h2 = l3.map((t21, e5) => {
          const n5 = e5, r5 = e5 + 1, a4 = c3(i4[n5], n5);
          return (c3(i4[r5], r5) + a4) / 2;
        }), f2 = r4 - t20.ringWidth - t20.ringInset, p2 = r4 - t20.ringInset, d2 = p2 - (p2 - f2) / 2;
        e4.append("g").attr("class", "label").attr("transform", n4).selectAll("text").data(l3).enter().append("text").attr("transform", (e5, n5) => {
          const i5 = h2[n5], a4 = `rotate(${i5}) translate(0, ${t20.labelInset - r4})`, o4 = `rotate(${i5}) translate(0, ${t20.labelInset / 2 - d2})`;
          return e5.position === "OUTSIDE" ? a4 : o4;
        }).text((t21) => t21.text || "").attr("class", "segment-value").style("text-anchor", "middle").style("font-size", (e5) => e5.fontSize || t20.labelFontSize).style("font-weight", "bold").style("fill", (e5) => e5.color || t20.textColor);
      }({ config: t19, svg: e3, centerTx: n3, r: r3, ticks: i3, tickData: a2, scale: o2, range: u2 });
    e3.append("g").attr("class", "label").attr("transform", n3).selectAll("text").data(i3).enter().append("text").attr("transform", (e4, n4) => {
      const i4 = t19.customSegmentStops.length === 0 ? o2(e4) : ru(a2, n4);
      return `rotate(${t19.minAngle + i4 * u2}) translate(0, ${t19.labelInset - r3})`;
    }).text(t19.labelFormat).attr("class", "segment-value").style("text-anchor", "middle").style("font-size", t19.labelFontSize).style("font-weight", "bold").style("fill", t19.textColor);
  }({ config: e2, svg: i2, centerTx: r2, r: n2 }), { current_value_text: xu({ config: e2, svg: i2 }), pointer: Mu({ config: e2, svg: i2, r: n2, centerTx: r2 }) };
};
function xu({ config: t18, svg: e2 }) {
  const n2 = (t18.width + 2 * t18.paddingHorizontal) / 2, r2 = (t18.width + 4 * t18.paddingVertical) / 2;
  return e2.append("g").attr("transform", `translate(${n2}, ${r2})`).append("text").attr("class", "current-value").attr("text-anchor", "middle").attr("y", 23).text(t18.currentValue).style("font-size", t18.valueTextFontSize).style("font-weight", t18.valueTextFontWeight).style("fill", t18.textColor);
}
function Mu({ config: t18, svg: e2, r: n2, centerTx: r2 }) {
  const i2 = function({ heightRatio: t19, radius: e3 }) {
    if (t19 < 0 || t19 > 1)
      throw new Error(`Invalid needleHeightRatio given - ${t19}`);
    return Math.round(e3 * t19);
  }({ heightRatio: t18.needleHeightRatio, radius: n2 }), a2 = [[t18.pointerWidth / 2, 0], [0, -i2], [-t18.pointerWidth / 2, 0], [0, t18.pointerTailLength], [t18.pointerWidth / 2, 0]], o2 = function() {
    var t19 = Ra, e3 = $a, n3 = da(true), r3 = null, i3 = Pa, a3 = null;
    function o3(o4) {
      var u2, s2, l2, c2 = o4.length, h2 = false;
      for (r3 == null && (a3 = i3(l2 = Oi())), u2 = 0; u2 <= c2; ++u2)
        !(u2 < c2 && n3(s2 = o4[u2], u2, o4)) === h2 && ((h2 = !h2) ? a3.lineStart() : a3.lineEnd()), h2 && a3.point(+t19(s2, u2, o4), +e3(s2, u2, o4));
      if (l2)
        return a3 = null, l2 + "" || null;
    }
    return o3.x = function(e4) {
      return arguments.length ? (t19 = typeof e4 == "function" ? e4 : da(+e4), o3) : t19;
    }, o3.y = function(t20) {
      return arguments.length ? (e3 = typeof t20 == "function" ? t20 : da(+t20), o3) : e3;
    }, o3.defined = function(t20) {
      return arguments.length ? (n3 = typeof t20 == "function" ? t20 : da(!!t20), o3) : n3;
    }, o3.curve = function(t20) {
      return arguments.length ? (i3 = t20, r3 != null && (a3 = i3(r3)), o3) : i3;
    }, o3.context = function(t20) {
      return arguments.length ? (t20 == null ? r3 = a3 = null : a3 = i3(r3 = t20), o3) : r3;
    }, o3;
  }().curve(Ba);
  return e2.append("g").data([a2]).attr("class", "pointer").attr("transform", r2).style("fill", t18.needleColor).append("path").attr("d", o2).attr("transform", `rotate(${t18.minAngle})`);
}
var Su = (() => {
  try {
    var t18 = zo(Object, "defineProperty");
    return t18({}, "", {}), t18;
  } catch (t19) {
  }
})(), Au = Su;
function Nu(t18, e2) {
  return t18 === e2 || t18 != t18 && e2 != e2;
}
function Cu(t18, e2) {
  for (var n2 = t18.length; n2--; )
    if (Nu(t18[n2][0], e2))
      return n2;
  return -1;
}
var Ou = Array.prototype.splice;
function ku(t18) {
  var e2 = -1, n2 = t18 == null ? 0 : t18.length;
  for (this.clear(); ++e2 < n2; ) {
    var r2 = t18[e2];
    this.set(r2[0], r2[1]);
  }
}
ku.prototype.clear = function() {
  this.__data__ = [], this.size = 0;
}, ku.prototype.delete = function(t18) {
  var e2 = this.__data__, n2 = Cu(e2, t18);
  return !(n2 < 0) && (n2 == e2.length - 1 ? e2.pop() : Ou.call(e2, n2, 1), --this.size, true);
}, ku.prototype.get = function(t18) {
  var e2 = this.__data__, n2 = Cu(e2, t18);
  return n2 < 0 ? void 0 : e2[n2][1];
}, ku.prototype.has = function(t18) {
  return Cu(this.__data__, t18) > -1;
}, ku.prototype.set = function(t18, e2) {
  var n2 = this.__data__, r2 = Cu(n2, t18);
  return r2 < 0 ? (++this.size, n2.push([t18, e2])) : n2[r2][1] = e2, this;
};
var ju = zo(Object, "create");
var Eu = Object.prototype.hasOwnProperty;
var Iu = Object.prototype.hasOwnProperty;
function Tu(t18) {
  var e2 = -1, n2 = t18 == null ? 0 : t18.length;
  for (this.clear(); ++e2 < n2; ) {
    var r2 = t18[e2];
    this.set(r2[0], r2[1]);
  }
}
function Lu(t18, e2) {
  var n2, r2, i2 = t18.__data__;
  return ((r2 = typeof (n2 = e2)) == "string" || r2 == "number" || r2 == "symbol" || r2 == "boolean" ? n2 !== "__proto__" : n2 === null) ? i2[typeof e2 == "string" ? "string" : "hash"] : i2.map;
}
function Pu(t18) {
  var e2 = -1, n2 = t18 == null ? 0 : t18.length;
  for (this.clear(); ++e2 < n2; ) {
    var r2 = t18[e2];
    this.set(r2[0], r2[1]);
  }
}
Tu.prototype.clear = function() {
  this.__data__ = ju ? ju(null) : {}, this.size = 0;
}, Tu.prototype.delete = function(t18) {
  var e2 = this.has(t18) && delete this.__data__[t18];
  return this.size -= e2 ? 1 : 0, e2;
}, Tu.prototype.get = function(t18) {
  var e2 = this.__data__;
  if (ju) {
    var n2 = e2[t18];
    return n2 === "__lodash_hash_undefined__" ? void 0 : n2;
  }
  return Eu.call(e2, t18) ? e2[t18] : void 0;
}, Tu.prototype.has = function(t18) {
  var e2 = this.__data__;
  return ju ? e2[t18] !== void 0 : Iu.call(e2, t18);
}, Tu.prototype.set = function(t18, e2) {
  var n2 = this.__data__;
  return this.size += this.has(t18) ? 0 : 1, n2[t18] = ju && e2 === void 0 ? "__lodash_hash_undefined__" : e2, this;
}, Pu.prototype.clear = function() {
  this.size = 0, this.__data__ = { hash: new Tu(), map: new (Vo || ku)(), string: new Tu() };
}, Pu.prototype.delete = function(t18) {
  var e2 = Lu(this, t18).delete(t18);
  return this.size -= e2 ? 1 : 0, e2;
}, Pu.prototype.get = function(t18) {
  return Lu(this, t18).get(t18);
}, Pu.prototype.has = function(t18) {
  return Lu(this, t18).has(t18);
}, Pu.prototype.set = function(t18, e2) {
  var n2 = Lu(this, t18), r2 = n2.size;
  return n2.set(t18, e2), this.size += n2.size == r2 ? 0 : 1, this;
};
function Ru(t18) {
  var e2 = this.__data__ = new ku(t18);
  this.size = e2.size;
}
Ru.prototype.clear = function() {
  this.__data__ = new ku(), this.size = 0;
}, Ru.prototype.delete = function(t18) {
  var e2 = this.__data__, n2 = e2.delete(t18);
  return this.size = e2.size, n2;
}, Ru.prototype.get = function(t18) {
  return this.__data__.get(t18);
}, Ru.prototype.has = function(t18) {
  return this.__data__.has(t18);
}, Ru.prototype.set = function(t18, e2) {
  var n2 = this.__data__;
  if (n2 instanceof ku) {
    var r2 = n2.__data__;
    if (!Vo || r2.length < 199)
      return r2.push([t18, e2]), this.size = ++n2.size, this;
    n2 = this.__data__ = new Pu(r2);
  }
  return n2.set(t18, e2), this.size = n2.size, this;
};
function $u(t18) {
  var e2 = -1, n2 = t18 == null ? 0 : t18.length;
  for (this.__data__ = new Pu(); ++e2 < n2; )
    this.add(t18[e2]);
}
function zu(t18, e2) {
  for (var n2 = -1, r2 = t18 == null ? 0 : t18.length; ++n2 < r2; )
    if (e2(t18[n2], n2, t18))
      return true;
  return false;
}
$u.prototype.add = $u.prototype.push = function(t18) {
  return this.__data__.set(t18, "__lodash_hash_undefined__"), this;
}, $u.prototype.has = function(t18) {
  return this.__data__.has(t18);
};
function qu(t18, e2, n2, r2, i2, a2) {
  var o2 = 1 & n2, u2 = t18.length, s2 = e2.length;
  if (u2 != s2 && !(o2 && s2 > u2))
    return false;
  var l2 = a2.get(t18), c2 = a2.get(e2);
  if (l2 && c2)
    return l2 == e2 && c2 == t18;
  var h2 = -1, f2 = true, p2 = 2 & n2 ? new $u() : void 0;
  for (a2.set(t18, e2), a2.set(e2, t18); ++h2 < u2; ) {
    var d2 = t18[h2], v2 = e2[h2];
    if (r2)
      var g2 = o2 ? r2(v2, d2, h2, e2, t18, a2) : r2(d2, v2, h2, t18, e2, a2);
    if (g2 !== void 0) {
      if (g2)
        continue;
      f2 = false;
      break;
    }
    if (p2) {
      if (!zu(e2, (t19, e3) => {
        if (o3 = e3, !p2.has(o3) && (d2 === t19 || i2(d2, t19, n2, r2, a2)))
          return p2.push(e3);
        var o3;
      })) {
        f2 = false;
        break;
      }
    } else if (d2 !== v2 && !i2(d2, v2, n2, r2, a2)) {
      f2 = false;
      break;
    }
  }
  return a2.delete(t18), a2.delete(e2), f2;
}
var Vu = r.Uint8Array;
function Uu(t18) {
  var e2 = -1, n2 = Array(t18.size);
  return t18.forEach((t19, r2) => {
    n2[++e2] = [r2, t19];
  }), n2;
}
function Fu(t18) {
  var e2 = -1, n2 = Array(t18.size);
  return t18.forEach((t19) => {
    n2[++e2] = t19;
  }), n2;
}
var Bu = s ? s.prototype : void 0, Du = Bu ? Bu.valueOf : void 0;
var Wu = Object.prototype.propertyIsEnumerable, Hu = Object.getOwnPropertySymbols, Xu = Hu ? (t18) => t18 == null ? [] : (t18 = Object(t18), function(t19, e2) {
  for (var n2 = -1, r2 = t19 == null ? 0 : t19.length, i2 = 0, a2 = []; ++n2 < r2; ) {
    var o2 = t19[n2];
    e2(o2, n2, t19) && (a2[i2++] = o2);
  }
  return a2;
}(Hu(t18), (e2) => Wu.call(t18, e2))) : function() {
  return [];
}, Yu = Xu;
function Qu(t18) {
  return function(t19, e2, n2) {
    var r2 = e2(t19);
    return R(t19) ? r2 : ((t20, e3) => {
      for (var n3 = -1, r3 = e3.length, i2 = t20.length; ++n3 < r3; )
        t20[i2 + n3] = e3[n3];
      return t20;
    })(r2, n2(t19));
  }(t18, ut, Yu);
}
var Gu = Object.prototype.hasOwnProperty;
var Zu = "[object Object]", Ju = Object.prototype.hasOwnProperty;
function Ku(t18, e2, n2, r2, i2, a2) {
  var o2 = R(t18), u2 = R(e2), s2 = o2 ? "[object Array]" : Go(t18), l2 = u2 ? "[object Array]" : Go(e2), c2 = (s2 = s2 == "[object Arguments]" ? Zu : s2) == Zu, h2 = (l2 = l2 == "[object Arguments]" ? Zu : l2) == Zu, f2 = s2 == l2;
  if (f2 && V(t18)) {
    if (!V(e2))
      return false;
    o2 = true, c2 = false;
  }
  if (f2 && !c2)
    return a2 || (a2 = new Ru()), o2 || Z(t18) ? qu(t18, e2, n2, r2, i2, a2) : function(t19, e3, n3, r3, i3, a3, o3) {
      switch (n3) {
        case "[object DataView]":
          if (t19.byteLength != e3.byteLength || t19.byteOffset != e3.byteOffset)
            return false;
          t19 = t19.buffer, e3 = e3.buffer;
        case "[object ArrayBuffer]":
          return !(t19.byteLength != e3.byteLength || !a3(new Vu(t19), new Vu(e3)));
        case "[object Boolean]":
        case "[object Date]":
        case "[object Number]":
          return Nu(+t19, +e3);
        case "[object Error]":
          return t19.name == e3.name && t19.message == e3.message;
        case "[object RegExp]":
        case "[object String]":
          return t19 == e3 + "";
        case "[object Map]":
          var u3 = Uu;
        case "[object Set]":
          var s3 = 1 & r3;
          if (u3 || (u3 = Fu), t19.size != e3.size && !s3)
            return false;
          var l3 = o3.get(t19);
          if (l3)
            return l3 == e3;
          r3 |= 2, o3.set(t19, e3);
          var c3 = qu(u3(t19), u3(e3), r3, i3, a3, o3);
          return o3.delete(t19), c3;
        case "[object Symbol]":
          if (Du)
            return Du.call(t19) == Du.call(e3);
      }
      return false;
    }(t18, e2, s2, n2, r2, i2, a2);
  if (!(1 & n2)) {
    var p2 = c2 && Ju.call(t18, "__wrapped__"), d2 = h2 && Ju.call(e2, "__wrapped__");
    if (p2 || d2) {
      var v2 = p2 ? t18.value() : t18, g2 = d2 ? e2.value() : e2;
      return a2 || (a2 = new Ru()), i2(v2, g2, n2, r2, a2);
    }
  }
  return !!f2 && (a2 || (a2 = new Ru()), function(t19, e3, n3, r3, i3, a3) {
    var o3 = 1 & n3, u3 = Qu(t19), s3 = u3.length;
    if (s3 != Qu(e3).length && !o3)
      return false;
    for (var l3 = s3; l3--; ) {
      var c3 = u3[l3];
      if (!(o3 ? c3 in e3 : Gu.call(e3, c3)))
        return false;
    }
    var h3 = a3.get(t19), f3 = a3.get(e3);
    if (h3 && f3)
      return h3 == e3 && f3 == t19;
    var p3 = true;
    a3.set(t19, e3), a3.set(e3, t19);
    for (var d3 = o3; ++l3 < s3; ) {
      var v3 = t19[c3 = u3[l3]], g3 = e3[c3];
      if (r3)
        var y2 = o3 ? r3(g3, v3, c3, e3, t19, a3) : r3(v3, g3, c3, t19, e3, a3);
      if (!(y2 === void 0 ? v3 === g3 || i3(v3, g3, n3, r3, a3) : y2)) {
        p3 = false;
        break;
      }
      d3 || (d3 = c3 == "constructor");
    }
    if (p3 && !d3) {
      var m2 = t19.constructor, _2 = e3.constructor;
      m2 == _2 || !("constructor" in t19) || !("constructor" in e3) || typeof m2 == "function" && m2 instanceof m2 && typeof _2 == "function" && _2 instanceof _2 || (p3 = false);
    }
    return a3.delete(t19), a3.delete(e3), p3;
  }(t18, e2, n2, r2, i2, a2));
}
function ts(t18, e2, n2, r2, i2) {
  return t18 === e2 || (t18 == null || e2 == null || !g(t18) && !g(e2) ? t18 != t18 && e2 != e2 : Ku(t18, e2, n2, r2, ts, i2));
}
function es(e2) {
  return e2 == e2 && !t(e2);
}
function ns(t18, e2) {
  return (n2) => n2 != null && (n2[t18] === e2 && (e2 !== void 0 || t18 in Object(n2)));
}
function rs(t18) {
  var e2 = function(t19) {
    for (var e3 = ut(t19), n2 = e3.length; n2--; ) {
      var r2 = e3[n2], i2 = t19[r2];
      e3[n2] = [r2, i2, es(i2)];
    }
    return e3;
  }(t18);
  return e2.length == 1 && e2[0][2] ? ns(e2[0][0], e2[0][1]) : (n2) => n2 === t18 || function(t19, e3, n3, r2) {
    var i2 = n3.length, a2 = i2, o2 = !r2;
    if (t19 == null)
      return !a2;
    for (t19 = Object(t19); i2--; ) {
      var u2 = n3[i2];
      if (o2 && u2[2] ? u2[1] !== t19[u2[0]] : !(u2[0] in t19))
        return false;
    }
    for (; ++i2 < a2; ) {
      var s2 = (u2 = n3[i2])[0], l2 = t19[s2], c2 = u2[1];
      if (o2 && u2[2]) {
        if (l2 === void 0 && !(s2 in t19))
          return false;
      } else {
        var h2 = new Ru();
        if (r2)
          var f2 = r2(l2, c2, s2, t19, e3, h2);
        if (!(f2 === void 0 ? ts(c2, l2, 3, r2, h2) : f2))
          return false;
      }
    }
    return true;
  }(n2, t18, e2);
}
var is = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, as = /^\w*$/;
function os(t18, e2) {
  if (R(t18))
    return false;
  var n2 = typeof t18;
  return !(n2 != "number" && n2 != "symbol" && n2 != "boolean" && t18 != null && !y(t18)) || (as.test(t18) || !is.test(t18) || e2 != null && t18 in Object(e2));
}
function us(t18, e2) {
  if (typeof t18 != "function" || e2 != null && typeof e2 != "function")
    throw new TypeError("Expected a function");
  var n2 = function() {
    var r2 = arguments, i2 = e2 ? e2.apply(this, r2) : r2[0], a2 = n2.cache;
    if (a2.has(i2))
      return a2.get(i2);
    var o2 = t18.apply(this, r2);
    return n2.cache = a2.set(i2, o2) || a2, o2;
  };
  return n2.cache = new (us.Cache || Pu)(), n2;
}
us.Cache = Pu;
var ss = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, ls = /\\(\\)?/g, cs = function(t18) {
  var e2 = us(t18, (t19) => (n2.size === 500 && n2.clear(), t19)), n2 = e2.cache;
  return e2;
}((t18) => {
  var e2 = [];
  return t18.charCodeAt(0) === 46 && e2.push(""), t18.replace(ss, (t19, n2, r2, i2) => {
    e2.push(r2 ? i2.replace(ls, "$1") : n2 || t19);
  }), e2;
}), hs = cs;
var fs = s ? s.prototype : void 0, ps = fs ? fs.toString : void 0;
function ds(t18) {
  if (typeof t18 == "string")
    return t18;
  if (R(t18))
    return function(t19, e3) {
      for (var n2 = -1, r2 = t19 == null ? 0 : t19.length, i2 = Array(r2); ++n2 < r2; )
        i2[n2] = e3(t19[n2], n2, t19);
      return i2;
    }(t18, ds) + "";
  if (y(t18))
    return ps ? ps.call(t18) : "";
  var e2 = t18 + "";
  return e2 == "0" && 1 / t18 == -Infinity ? "-0" : e2;
}
function vs(t18, e2) {
  return R(t18) ? t18 : os(t18, e2) ? [t18] : hs(function(t19) {
    return t19 == null ? "" : ds(t19);
  }(t18));
}
function gs(t18) {
  if (typeof t18 == "string" || y(t18))
    return t18;
  var e2 = t18 + "";
  return e2 == "0" && 1 / t18 == -Infinity ? "-0" : e2;
}
function ys(t18, e2) {
  for (var n2 = 0, r2 = (e2 = vs(e2, t18)).length; t18 != null && n2 < r2; )
    t18 = t18[gs(e2[n2++])];
  return n2 && n2 == r2 ? t18 : void 0;
}
function ms(t18, e2) {
  return t18 != null && e2 in Object(t18);
}
function _s(t18, e2) {
  return t18 != null && function(t19, e3, n2) {
    for (var r2 = -1, i2 = (e3 = vs(e3, t19)).length, a2 = false; ++r2 < i2; ) {
      var o2 = gs(e3[r2]);
      if (!(a2 = t19 != null && n2(t19, o2)))
        break;
      t19 = t19[o2];
    }
    return a2 || ++r2 != i2 ? a2 : !!(i2 = t19 == null ? 0 : t19.length) && B(i2) && F(o2, i2) && (R(t19) || P(t19));
  }(t18, e2, ms);
}
function bs(t18, e2) {
  return os(t18) && es(e2) ? ns(gs(t18), e2) : (n2) => {
    var r2 = function(t19, e3, n3) {
      var r3 = t19 == null ? void 0 : ys(t19, e3);
      return r3 === void 0 ? n3 : r3;
    }(n2, t18);
    return r2 === void 0 && r2 === e2 ? _s(n2, t18) : ts(e2, r2, 3);
  };
}
function ws(t18) {
  return os(t18) ? (e2 = gs(t18), (t19) => t19 == null ? void 0 : t19[e2]) : function(t19) {
    return (e3) => ys(e3, t19);
  }(t18);
  var e2;
}
const xs = { value: { type: Number, required: true }, minValue: { type: Number, required: true }, maxValue: { type: Number, required: true }, forceRender: { type: Boolean, required: true }, width: { type: Number, required: true }, height: { type: Number, required: true }, paddingHorizontal: { type: Number, required: true }, paddingVertical: { type: Number, required: true }, dimensionUnit: { type: String, required: true }, fluidWidth: { type: Boolean, required: true }, segments: { type: Number, required: true }, maxSegmentLabels: { type: Number }, customSegmentStops: { type: Array }, customSegmentLabels: { type: Array }, needleColor: { type: String, required: true }, startColor: { type: String, required: true }, endColor: { type: String, required: true }, segmentColors: { type: Array, required: true }, needleTransition: { type: String, required: true }, needleTransitionDuration: { type: Number, required: true }, needleHeightRatio: { type: Number, required: true }, ringWidth: { type: Number, required: true }, textColor: { type: String, required: true }, valueFormat: { type: String, required: true }, currentValueText: { type: String, required: true }, currentValuePlaceholderStyle: { type: String, required: true }, labelFontSize: { type: String, required: true }, valueTextFontSize: { type: String, required: true }, valueTextFontWeight: { type: String, required: true }, svgAriaLabel: { type: String, required: true } }, Ms = function(t18, e2) {
  var n2, r2 = {};
  return e2 = typeof (n2 = e2) == "function" ? n2 : n2 == null ? ht : typeof n2 == "object" ? R(n2) ? bs(n2[0], n2[1]) : rs(n2) : ws(n2), st(t18, (t19, n3, i2) => {
    !function(t20, e3, n4) {
      e3 == "__proto__" && Au ? Au(t20, e3, { configurable: true, enumerable: true, value: n4, writable: true }) : t20[e3] = n4;
    }(r2, n3, e2(t19, n3, i2));
  }), r2;
}(xs, (t18, e2) => ({ ...t18, default: { customSegmentStops: () => [], customSegmentLabels: () => [], segmentColors: () => [], svgAriaLabel: () => "Vue Speedometer" }[e2] || hu[e2], required: false }));
function Ss(t18) {
  const e2 = ut(xs);
  let n2 = {};
  return pt(e2, (e3) => {
    n2[e3] = t18[e3];
  }), n2;
}
var As = { name: "vue-speedometer", props: Ms, template: "\n    <div></div>\n  ", beforeCreate: function() {
  this.d3_refs = { pointer: false, current_value_text: false };
}, created: function() {
  this._initWatchers();
}, mounted: function() {
  this.renderGauge();
}, data: function() {
  return { force_render: this.forceRender };
}, watch: { forceRender: function(t18) {
  this.force_render = t18;
} }, methods: { _initWatchers() {
  this.updateReadings = A(this._updateReadings, 0, { leading: false, trailing: true });
  pt(ut(xs).filter((t18) => t18 !== "forceRender"), (t18) => {
    this.$watch(t18, function(e2, n2) {
      this.watchUpdater({ newValue: e2, oldValue: n2, prop: t18 });
    });
  });
}, watchUpdater: function({ newValue: t18, oldValue: e2, prop: n2 }) {
  const r2 = Object.assign({ [n2]: t18 || e2 }, n2 === "value" ? { value: t18 } : {});
  var i2, a2;
  this.config = (i2 = this.config, a2 = r2, { ...i2, ...a2 }), this.$nextTick().then(() => {
    this.updateReadings();
  });
}, renderGauge: function() {
  this.config = (({ PROPS: t18, parentWidth: e2, parentHeight: n2 }) => {
    const r2 = { width: t18.fluidWidth ? e2 : t18.width, height: t18.fluidWidth ? n2 : t18.height, paddingHorizontal: t18.paddingHorizontal, paddingVertical: t18.paddingVertical, dimensionUnit: t18.dimensionUnit, ringWidth: t18.ringWidth, minValue: t18.minValue, maxValue: t18.maxValue, needleColor: t18.needleColor, majorTicks: t18.segments, customSegmentStops: t18.customSegmentStops, customSegmentLabels: t18.customSegmentLabels, maxSegmentLabels: au({ maxSegmentLabelCount: t18.maxSegmentLabels, segmentCount: t18.segments }), segmentColors: t18.segmentColors, arcColorFn: t18.segmentColors.length > 0 ? pa(t18.segmentColors) : Bn(on(t18.startColor), on(t18.endColor)), needleTransition: t18.needleTransition, needleTransitionDuration: t18.needleTransitionDuration, needleHeightRatio: t18.needleHeightRatio, textColor: t18.textColor, labelFormat: Bi(t18.valueFormat), currentValueText: t18.currentValueText, currentValuePlaceholderStyle: t18.currentValuePlaceholderStyle, labelFontSize: t18.labelFontSize, valueTextFontSize: t18.valueTextFontSize, valueTextFontWeight: t18.valueTextFontWeight, svgAriaLabel: t18.svgAriaLabel };
    return Object.assign({}, fu, r2);
  })({ PROPS: Ss(this), parentWidth: this.$el.parentNode.clientWidth, parentHeight: this.$el.parentNode.clientHeight }), $e(this.$el).select("svg").remove(), this.d3_refs = wu({ container: this.$el, config: this.config }), bu({ d3_refs: this.d3_refs, newValue: this.value, config: this.config });
}, _updateReadings: function() {
  this.force_render ? this.renderGauge() : bu({ d3_refs: this.d3_refs, newValue: this.config.value || 0, config: this.config });
} } };
export { lu as CustomSegmentLabelPosition, cu as Transition, As as default };
